// @mercurjs/b2c-core/.medusa/server/src/api/vendor/fulfillment-sets/[id]/service-zones/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const utils_2 = require("../../../../../shared/infra/http/utils");
const fulfillment_set_1 = require("../../../../../workflows/fulfillment-set");
/**
 * @oas [post] /vendor/fulfillment-sets/{id}/service-zones
 * operationId: "VendorCreateServiceZone"
 * summary: "Create a Service Zone"
 * description: "Creates a Service Zone."
 * x-authenticated: true
 * parameters:
 *   - in: path
 *     name: id
 *     required: true
 *     description: The ID of the Fulfillment Set.
 *     schema:
 *       type: string
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateServiceZone"
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             fulfillment_set:
 *               $ref: "#/components/schemas/VendorFulfillmentSet"
 * tags:
 *   - Vendor Fulfillment Sets
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const seller = await (0, utils_2.fetchSellerByAuthActorId)(req.auth_context.actor_id, req.scope);
    await (0, fulfillment_set_1.createVendorServiceZonesWorkflow)(req.scope).run({
        input: {
            data: [{ fulfillment_set_id: req.params.id, ...req.validatedBody }],
            seller_id: seller.id,
        },
    });
    const { data: [fulfillmentSet], } = await query.graph({
        entity: "fulfillment_set",
        fields: req.queryConfig.fields,
        filters: {
            id: req.params.id,
        },
    }, { throwIfKeyNotFound: true });
    res.json({ fulfillment_set: fulfillmentSet });
};
exports.POST = POST;
