// @mercurjs/b2c-core/.medusa/server/src/api/vendor/promotions/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorPromotionsMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const seller_promotion_1 = __importDefault(require("../../../links/seller-promotion"));
const middlewares_1 = require("../../../shared/infra/http/middlewares");
const query_config_1 = require("./query-config");
const validators_1 = require("./validators");
const vendor_promotions_rule_value_options_path_params_guard_1 = require("../../../shared/infra/http/middlewares/vendor-promotions-rule-value-options-path-params-guard");
exports.vendorPromotionsMiddlewares = [
    {
        method: ["GET"],
        matcher: "/vendor/promotions",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.list),
            (0, middlewares_1.filterBySellerId)(),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/promotions/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/promotions/:id/:rule_type",
        middlewares: [
            (0, framework_1.unlessPath)(/.*\/promotions\/rule-attribute-options/, (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionRuleTypeParams, query_config_1.vendorPromotionQueryConfig.retrieve)),
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/promotions/:id",
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_1.VendorUpdatePromotion),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
        ],
    },
    {
        method: ["DELETE"],
        matcher: "/vendor/promotions/:id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/promotions",
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_1.VendorCreatePromotion),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/promotions/:id/buy-rules/batch",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_1.VendorBatchPromotionRules),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/promotions/:id/rules/batch",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_1.VendorBatchPromotionRules),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/promotions/:id/target-rules/batch",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_promotion_1.default.entryPoint,
                filterField: "promotion_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_1.VendorBatchPromotionRules),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsParams, query_config_1.vendorPromotionQueryConfig.retrieve),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/promotions/rule-value-options/:rule_type/:rule_attribute_id",
        middlewares: [
            vendor_promotions_rule_value_options_path_params_guard_1.vendorPromotionsRuleValueOptionsPathParamsGuard,
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionsRuleValueParams, query_config_1.listRuleValueTransformQueryConfig),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/promotions/rule-attribute-options/:rule_type",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetPromotionRuleParams, query_config_1.vendorRuleTransformQueryConfig.list),
        ],
    },
];
