// @mercurjs/b2c-core/.medusa/server/src/api/vendor/product-collections/query-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorProductCollectionsProductsQueryConfig = exports.vendorProductCollectionsProductsFields = exports.vendorProductCollectionQueryConfig = exports.vendorProductCollectionFields = void 0;
exports.vendorProductCollectionFields = [
    'id',
    'title',
    'handle',
    'created_at',
    'updated_at',
    'metadata'
];
exports.vendorProductCollectionQueryConfig = {
    list: {
        defaults: exports.vendorProductCollectionFields,
        isList: true
    },
    retrieve: {
        defaults: exports.vendorProductCollectionFields,
        isList: false
    }
};
exports.vendorProductCollectionsProductsFields = [
    'id',
    'title',
    'status',
    'thumbnail',
    'handle',
    'collection.id',
    'collection.title',
    'variants.id'
];
exports.vendorProductCollectionsProductsQueryConfig = {
    list: {
        defaults: exports.vendorProductCollectionsProductsFields,
        isList: true
    },
    retrieve: {
        defaults: exports.vendorProductCollectionsProductsFields,
        isList: false
    }
};
