// @mercurjs/b2c-core/.medusa/server/src/api/vendor/campaigns/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorCampaignsMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const validators_1 = require("@medusajs/medusa/api/utils/validators");
const seller_campaign_1 = __importDefault(require("../../../links/seller-campaign"));
const middlewares_1 = require("../../../shared/infra/http/middlewares");
const query_config_1 = require("./query-config");
const validators_2 = require("./validators");
exports.vendorCampaignsMiddlewares = [
    {
        method: ['GET'],
        matcher: '/vendor/campaigns',
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetCampaignsParams, query_config_1.vendorCampaignQueryConfig.list),
            (0, middlewares_1.filterBySellerId)()
        ]
    },
    {
        method: ['GET'],
        matcher: '/vendor/campaigns/:id',
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetCampaignByIdParams, query_config_1.vendorCampaignQueryConfig.retrieve),
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_campaign_1.default.entryPoint,
                filterField: 'campaign_id'
            })
        ]
    },
    {
        method: ['POST'],
        matcher: '/vendor/campaigns/:id',
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_2.VendorUpdateCampaign),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetCampaignsParams, query_config_1.vendorCampaignQueryConfig.retrieve),
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_campaign_1.default.entryPoint,
                filterField: 'campaign_id'
            })
        ]
    },
    {
        method: ['DELETE'],
        matcher: '/vendor/campaigns/:id',
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_campaign_1.default.entryPoint,
                filterField: 'campaign_id'
            })
        ]
    },
    {
        method: ['POST'],
        matcher: '/vendor/campaigns',
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_2.VendorCreateCampaign),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetCampaignsParams, query_config_1.vendorCampaignQueryConfig.retrieve)
        ]
    },
    {
        method: ["POST"],
        matcher: "/vendor/campaigns/:id/promotions",
        middlewares: [
            (0, framework_1.validateAndTransformBody)((0, validators_1.createLinkBody)()),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetCampaignsParams, query_config_1.vendorCampaignQueryConfig.retrieve),
        ],
    },
];
