// @mercurjs/b2c-core/.medusa/server/src/api/vendor/campaigns/route.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = exports.GET = void 0;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const workflows_1 = require("../../../workflows/campaigns/workflows");
const seller_campaign_1 = __importDefault(require("../../../links/seller-campaign"));
/**
 * @oas [get] /vendor/campaigns
 * operationId: "VendorListCampaigns"
 * summary: "List Campaigns"
 * description: "Retrieves a list of campaigns for the authenticated vendor."
 * x-authenticated: true
 * parameters:
 *   - name: q
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: Search query to filter campaigns by name (case-insensitive).
 *   - name: offset
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to skip before starting to collect the result set.
 *   - name: limit
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to return.
 *   - name: fields
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: Comma-separated fields to include in the response.
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             campaigns:
 *               type: array
 *               items:
 *                 $ref: "#/components/schemas/VendorCampaign"
 *             count:
 *               type: integer
 *               description: The total number of items available
 *             offset:
 *               type: integer
 *               description: The number of items skipped before these items
 *             limit:
 *               type: integer
 *               description: The number of items per page
 * tags:
 *   - Vendor Campaigns
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const GET = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: relations, metadata } = await query.graph({
        entity: seller_campaign_1.default.entryPoint,
        fields: req.queryConfig.fields.map((field) => `campaign.${field}`),
        filters: {
            ...req.filterableFields,
            deleted_at: {
                $eq: null,
            },
        },
        pagination: req.queryConfig.pagination,
    });
    const activeCampaigns = relations
        .map((relation) => relation.campaign)
        .filter((campaign) => !!campaign);
    res.json({
        campaigns: activeCampaigns,
        count: activeCampaigns.length,
        offset: metadata?.skip,
        limit: metadata?.take,
    });
};
exports.GET = GET;
/**
 * @oas [post] /vendor/campaigns
 * operationId: "VendorCreateCampaign"
 * summary: "Create campaign"
 * description: "Creates a new campaign for the authenticated vendor."
 * x-authenticated: true
 * parameters:
 *   - name: fields
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: Comma-separated fields to include in the response.
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateCampaign"
 * responses:
 *   "201":
 *     description: Created
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             campaign:
 *               $ref: "#/components/schemas/VendorCampaign"
 * tags:
 *   - Vendor Campaigns
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const seller = await (0, framework_1.fetchSellerByAuthActorId)(req.auth_context?.actor_id, req.scope);
    const { result } = await workflows_1.createVendorCampaignWorkflow.run({
        container: req.scope,
        input: { campaign: req.validatedBody, seller_id: seller.id },
    });
    const { data: [campaign], } = await query.graph({
        entity: "campaign",
        fields: req.queryConfig.fields,
        filters: {
            id: result[0].id,
        },
    });
    res.status(201).json({ campaign });
};
exports.POST = POST;
