// @mercurjs/b2c-core/.medusa/server/src/api/vendor/campaigns/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorUpdateCampaign = exports.VendorCreateCampaign = exports.VendorCreateCampaignBudget = exports.VendorGetCampaignByIdParams = exports.VendorGetCampaignByIdParamsFields = exports.VendorGetCampaignsParams = exports.VendorGetCampaignsParamsFields = void 0;
const zod_1 = require("zod");
const utils_1 = require("@medusajs/framework/utils");
const validators_1 = require("@medusajs/medusa/api/utils/validators");
exports.VendorGetCampaignsParamsFields = zod_1.z.object({
    q: zod_1.z.string().optional()
});
exports.VendorGetCampaignsParams = (0, validators_1.createFindParams)({
    limit: 50,
    offset: 0
});
// TODO: will be used when we'll get back to using index module
// .merge(VendorGetCampaignsParamsFields)
// .strict()
const dateFilterSchema = zod_1.z
    .preprocess((val) => {
    if (typeof val === 'string') {
        try {
            return JSON.parse(val);
        }
        catch {
            return val;
        }
    }
    return val;
}, zod_1.z
    .object({
    $gte: zod_1.z.string().optional(),
    $lte: zod_1.z.string().optional(),
    $gt: zod_1.z.string().optional(),
    $lt: zod_1.z.string().optional(),
    $eq: zod_1.z.string().optional(),
    $ne: zod_1.z.string().optional()
})
    .optional())
    .optional();
exports.VendorGetCampaignByIdParamsFields = zod_1.z.object({
    q: zod_1.z.string().optional(),
    created_at: dateFilterSchema,
    updated_at: dateFilterSchema
});
exports.VendorGetCampaignByIdParams = (0, validators_1.createFindParams)({
    limit: 50,
    offset: 0
})
    .merge(exports.VendorGetCampaignByIdParamsFields)
    .strict();
/**
 * @schema VendorCreateCampaignBudget
 * type: object
 * properties:
 *   type:
 *     type: string
 *     enum: [spend,usage]
 *     description: The budget's type.
 *   limit:
 *     type: number
 *     description: The buget's limit.
 *   currency_code:
 *     type: string
 *     description: The budget's currency_code.
 */
exports.VendorCreateCampaignBudget = zod_1.z
    .object({
    type: zod_1.z.nativeEnum(utils_1.CampaignBudgetType),
    limit: zod_1.z.number().nullish(),
    currency_code: zod_1.z.string().nullish()
})
    .strict()
    .refine((data) => data.type !== utils_1.CampaignBudgetType.SPEND || (0, utils_1.isPresent)(data.currency_code), {
    path: ['currency_code'],
    message: `currency_code is required when budget type is ${utils_1.CampaignBudgetType.SPEND}`
})
    .refine((data) => data.type !== utils_1.CampaignBudgetType.USAGE || !(0, utils_1.isPresent)(data.currency_code), {
    path: ['currency_code'],
    message: `currency_code should not be present when budget type is ${utils_1.CampaignBudgetType.USAGE}`
});
exports.VendorCreateCampaign = zod_1.z
    .object({
    name: zod_1.z.string(),
    campaign_identifier: zod_1.z.string(),
    description: zod_1.z.string().nullish(),
    budget: exports.VendorCreateCampaignBudget.nullish(),
    starts_at: zod_1.z.coerce.date().nullish(),
    ends_at: zod_1.z.coerce.date().nullish()
})
    .strict();
exports.VendorUpdateCampaign = zod_1.z.object({
    name: zod_1.z.string().optional(),
    campaign_identifier: zod_1.z.string().optional(),
    description: zod_1.z.string().nullish(),
    budget: zod_1.z
        .object({
        limit: zod_1.z.number().nullish()
    })
        .optional(),
    starts_at: zod_1.z.coerce.date().nullish(),
    ends_at: zod_1.z.coerce.date().nullish()
});
