// @mercurjs/b2c-core/.medusa/server/src/workflows/price-list/workflows/create-vendor-price-list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createVendorPriceListWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const seller_1 = require("../../../modules/seller");
const steps_1 = require("../steps");
exports.createVendorPriceListWorkflow = (0, workflows_sdk_1.createWorkflow)("create-vendor-price-list", function ({ price_lists_data, seller_id, }) {
    (0, steps_1.validateVendorPriceListPricesStep)({
        create: price_lists_data.prices,
        seller_id,
    });
    const result = core_flows_1.createPriceListsWorkflow.runAsStep({
        input: {
            price_lists_data: [price_lists_data],
        },
    });
    const links = (0, workflows_sdk_1.transform)({ result, seller_id }, ({ result, seller_id }) => {
        return result.map((list) => {
            return {
                [seller_1.SELLER_MODULE]: {
                    seller_id: seller_id,
                },
                [utils_1.Modules.PRICING]: {
                    price_list_id: list.id,
                },
            };
        });
    });
    (0, core_flows_1.createRemoteLinkStep)(links);
    return new workflows_sdk_1.WorkflowResponse(result);
});
