// @mercurjs/requests/.medusa/server/src/workflows/requests/workflows/create-product-request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createProductRequestWorkflow = void 0;
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const requests_1 = require("../../../modules/requests");
const steps_1 = require("../steps");
const framework_2 = require("@mercurjs/framework");
exports.createProductRequestWorkflow = (0, workflows_sdk_1.createWorkflow)("create-product-request", function ({ data, seller_id, }) {
    const request = (0, steps_1.createRequestStep)(data);
    const link = (0, workflows_sdk_1.transform)({ request, seller_id }, ({ request, seller_id }) => {
        return [
            {
                [framework_1.SELLER_MODULE]: {
                    seller_id,
                },
                [requests_1.REQUESTS_MODULE]: {
                    request_id: request[0].id,
                },
            },
        ];
    });
    (0, core_flows_1.createRemoteLinkStep)(link);
    (0, framework_2.emitMultipleEventsStep)([
        {
            name: framework_1.RequestUpdated.CREATED,
            data: { ...data, sellerId: seller_id },
        },
        {
            name: framework_1.ProductRequestUpdatedEvent.CREATED,
            data: { id: request[0].id },
        },
    ]);
    const productRequestCreatedHook = (0, workflows_sdk_1.createHook)("productRequestCreated", {
        requestId: request[0].id,
        sellerId: seller_id,
    });
    return new workflows_sdk_1.WorkflowResponse(request, {
        hooks: [productRequestCreatedHook],
    });
});
