// @mercurjs/requests/.medusa/server/src/workflows/requests/workflows/create-request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRequestWorkflow = void 0;
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const requests_1 = require("../../../modules/requests");
const steps_1 = require("../steps");
exports.createRequestWorkflow = (0, workflows_sdk_1.createWorkflow)("create-request", function (input) {
    const request = (0, steps_1.createRequestStep)(input.data);
    const link = (0, workflows_sdk_1.transform)({ request, input }, ({ request, input }) => {
        return [
            {
                [framework_1.SELLER_MODULE]: {
                    seller_id: input.seller_id,
                },
                [requests_1.REQUESTS_MODULE]: {
                    request_id: request[0].id,
                },
            },
        ];
    });
    (0, core_flows_1.createRemoteLinkStep)(link);
    (0, core_flows_1.emitEventStep)({
        eventName: framework_1.RequestUpdated.CREATED,
        data: input.data,
    });
    const requestCreatedHook = (0, workflows_sdk_1.createHook)("requestCreated", {
        requestId: request[0].id,
        sellerId: input.seller_id,
    });
    return new workflows_sdk_1.WorkflowResponse(request, {
        hooks: [requestCreatedHook],
    });
});
