// @mercurjs/b2c-core/.medusa/server/src/workflows/promotions/workflows/create-vendor-promotion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createVendorPromotionWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const seller_1 = require("../../../modules/seller");
const steps_1 = require("../steps");
exports.createVendorPromotionWorkflow = (0, workflows_sdk_1.createWorkflow)("create-vendor-promotion", function (input) {
    (0, steps_1.verifyVendorCampaignStep)(input);
    (0, steps_1.verifyVendorPromotionStep)(input);
    (0, steps_1.verifyVendorTargetPromotionRulesStep)((0, workflows_sdk_1.transform)(input, (input) => ({
        rules: input.promotion.application_method.target_rules,
        seller_id: input.seller_id,
    })));
    const promotions = core_flows_1.createPromotionsWorkflow.runAsStep({
        input: {
            promotionsData: [input.promotion],
        },
    });
    const links = (0, workflows_sdk_1.transform)({ input, promotions }, ({ input, promotions }) => {
        const promo = promotions[0];
        const link = [
            {
                [seller_1.SELLER_MODULE]: {
                    seller_id: input.seller_id,
                },
                [utils_1.Modules.PROMOTION]: {
                    promotion_id: promo.id,
                },
            },
        ];
        if (promo.campaign) {
            link.push({
                [seller_1.SELLER_MODULE]: {
                    seller_id: input.seller_id,
                },
                [utils_1.Modules.PROMOTION]: {
                    campaign_id: promo.campaign.id,
                },
            });
        }
        return link;
    });
    (0, core_flows_1.createRemoteLinkStep)(links);
    return new workflows_sdk_1.WorkflowResponse(promotions);
});
