// @mercurjs/b2c-core/.medusa/server/src/api/vendor/customer-groups/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorLinkCustomersToGroup = exports.VendorCreateCustomerGroup = exports.VendorGetCustomerGroupsParams = exports.VendorGetCustomerGroupsParamsFields = exports.VendorCustomerInGroupFilters = exports.VendorGetCustomerGroupParams = void 0;
const zod_1 = require("zod");
const validators_1 = require("@medusajs/medusa/api/utils/validators");
exports.VendorGetCustomerGroupParams = (0, validators_1.createSelectParams)();
exports.VendorCustomerInGroupFilters = zod_1.z.object({
    id: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    email: zod_1.z
        .union([zod_1.z.string(), zod_1.z.array(zod_1.z.string()), (0, validators_1.createOperatorMap)()])
        .optional(),
    default_billing_address_id: zod_1.z
        .union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())])
        .optional(),
    default_shipping_address_id: zod_1.z
        .union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())])
        .optional(),
    company_name: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    first_name: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    last_name: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    created_by: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    created_at: (0, validators_1.createOperatorMap)().optional(),
    updated_at: (0, validators_1.createOperatorMap)().optional(),
    deleted_at: (0, validators_1.createOperatorMap)().optional()
});
exports.VendorGetCustomerGroupsParamsFields = zod_1.z.object({
    q: zod_1.z.string().optional(),
    id: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    name: zod_1.z
        .union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())])
        .optional()
        .or((0, validators_1.createOperatorMap)()),
    customers: zod_1.z
        .union([zod_1.z.string(), zod_1.z.array(zod_1.z.string()), exports.VendorCustomerInGroupFilters])
        .optional(),
    created_by: zod_1.z.union([zod_1.z.string(), zod_1.z.array(zod_1.z.string())]).optional(),
    created_at: (0, validators_1.createOperatorMap)().optional(),
    updated_at: (0, validators_1.createOperatorMap)().optional(),
    deleted_at: (0, validators_1.createOperatorMap)().optional()
});
exports.VendorGetCustomerGroupsParams = (0, validators_1.createFindParams)({
    offset: 0,
    limit: 50
});
exports.VendorCreateCustomerGroup = zod_1.z
    .object({
    name: zod_1.z.string()
})
    .strict();
exports.VendorLinkCustomersToGroup = zod_1.z
    .object({
    add: zod_1.z.array(zod_1.z.string()).default([]),
    remove: zod_1.z.array(zod_1.z.string()).default([])
})
    .strict();
