// @mercurjs/reviews/.medusa/server/src/workflows/review/workflows/create-review.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createReviewWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const framework_1 = require("@mercurjs/framework");
const reviews_1 = require("../../../modules/reviews");
const SELLER_MODULE = "seller";
const steps_1 = require("../steps");
exports.createReviewWorkflow = (0, workflows_sdk_1.createWorkflow)({
    name: "create-review",
}, function (input) {
    (0, steps_1.validateReviewStep)(input);
    const review = (0, steps_1.createReviewStep)(input);
    const link = (0, workflows_sdk_1.transform)({ input, review }, ({ input, review }) => {
        return input.reference === "product"
            ? [
                {
                    [utils_1.Modules.PRODUCT]: {
                        product_id: input.reference_id,
                    },
                    [reviews_1.REVIEW_MODULE]: {
                        review_id: review.id,
                    },
                },
            ]
            : [
                {
                    [SELLER_MODULE]: {
                        seller_id: input.reference_id,
                    },
                    [reviews_1.REVIEW_MODULE]: {
                        review_id: review.id,
                    },
                },
            ];
    });
    (0, core_flows_1.createRemoteLinkStep)(link);
    (0, core_flows_1.emitEventStep)({
        eventName: framework_1.AlgoliaEvents.REVIEW_CHANGED,
        data: { review },
    });
    const reviewCreatedHook = (0, workflows_sdk_1.createHook)("reviewCreated", {
        review_id: review.id,
    });
    return new workflows_sdk_1.WorkflowResponse(review, {
        hooks: [reviewCreatedHook],
    });
});
