// @mercurjs/reviews/.medusa/server/src/api/vendor/sellers/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorSellersMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const seller_review_1 = __importDefault(require("../../../links/seller-review"));
const framework_2 = require("@mercurjs/framework");
const query_config_1 = require("./query-config");
const validators_1 = require("./validators");
exports.vendorSellersMiddlewares = [
    {
        method: ["GET"],
        matcher: "/vendor/sellers/me/reviews",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetReviewsParams, query_config_1.vendorReviewQueryConfig.list),
            (0, framework_2.filterBySellerId)(),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/sellers/me/reviews/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetReviewsParams, query_config_1.vendorReviewQueryConfig.retrieve),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_review_1.default.entryPoint,
                filterField: "review_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/sellers/me/reviews/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetReviewsParams, query_config_1.vendorReviewQueryConfig.retrieve),
            (0, framework_1.validateAndTransformBody)(validators_1.VendorUpdateReview),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_review_1.default.entryPoint,
                filterField: "review_id",
            }),
        ],
    },
];
