// @mercurjs/b2c-core/.medusa/server/src/api/vendor/sellers/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
/**
 * @oas [post] /vendor/sellers
 * operationId: "VendorCreateSeller"
 * summary: "Create a Seller"
 * description: "Creates a request to create a new seller with an initial owner member."
 * x-authenticated: true
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateSeller"
 * responses:
 *   "201":
 *     description: Created
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             request:
 *               $ref: "#/components/schemas/VendorRequest"
 * tags:
 *   - Vendor Sellers
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    if (req.auth_context?.actor_id) {
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.INVALID_DATA, "Request already authenticated as a seller.");
    }
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { member, ...sellerData } = req.validatedBody;
    const { data: [identity], } = await query.graph({
        entity: "provider_identity",
        fields: ["id", "entity_id"],
        filters: {
            auth_identity_id: req.auth_context?.auth_identity_id,
        },
    });
    const { data: [existingRequest], } = await query.graph({
        entity: "request",
        fields: ["id"],
        filters: {
            submitter_id: identity.id,
            type: "seller",
        },
    });
    if (existingRequest) {
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.CONFLICT, "Request already exists!");
    }
    const eventBus = req.scope.resolve(utils_1.Modules.EVENT_BUS);
    await eventBus.emit({
        name: framework_1.SellerRequest.TO_CREATE,
        data: {
            data: {
                seller: { ...sellerData, email: sellerData.email || member.email },
                member,
                auth_identity_id: req.auth_context?.auth_identity_id,
                provider_identity_id: identity.entity_id,
            },
            type: "seller",
            submitter_id: identity.id,
        },
    });
    res.status(201).json({ ok: true });
};
exports.POST = POST;
