// @medusajs/dashboard/dist/order-list-4N3OV2PU.mjs
import {
  useOrderTableColumns
} from "./chunk-QXXHLVRQ.mjs";
import "./chunk-7DXVXBSA.mjs";
import {
  ConfigurableDataTable,
  createTableAdapter,
  orderColumnAdapter
} from "./chunk-W77EPLUD.mjs";
import "./chunk-6CPMXY5N.mjs";
import "./chunk-3I6G3NRU.mjs";
import "./chunk-SJ3E4T6B.mjs";
import "./chunk-INE2QCSC.mjs";
import "./chunk-XLDQPLK4.mjs";
import "./chunk-IQBAUTU5.mjs";
import "./chunk-ADOCJB6L.mjs";
import "./chunk-X6BAAGCL.mjs";
import "./chunk-P3UUX2T6.mjs";
import {
  _DataTable,
  useDataTable
} from "./chunk-PJ5EUKOB.mjs";
import "./chunk-HQKGZADC.mjs";
import "./chunk-EMIHDNB7.mjs";
import "./chunk-DK4WIVY6.mjs";
import {
  useOrderTableQuery
} from "./chunk-IW4TACOD.mjs";
import "./chunk-FKIOYL3D.mjs";
import {
  SingleColumnPage
} from "./chunk-GIZFNLKK.mjs";
import {
  useDataTableDateFilters
} from "./chunk-Q33G337Z.mjs";
import "./chunk-GRNWRTT3.mjs";
import "./chunk-C76H5USB.mjs";
import "./chunk-PFKKVLZX.mjs";
import "./chunk-MNXC6Q4F.mjs";
import "./chunk-DG7J63J2.mjs";
import {
  useFeatureFlag
} from "./chunk-RM6NY572.mjs";
import {
  useExtension
} from "./chunk-C5P5PL3E.mjs";
import "./chunk-LPEUYMRK.mjs";
import "./chunk-OZPB6JBL.mjs";
import "./chunk-OC7BQLYI.mjs";
import "./chunk-S4DMV3ZT.mjs";
import "./chunk-67ORSRVT.mjs";
import {
  useOrders
} from "./chunk-WTK7SR6S.mjs";
import "./chunk-6XWHQ6ON.mjs";
import {
  useSalesChannels
} from "./chunk-OGAZYXOV.mjs";
import "./chunk-KX4P5L3R.mjs";
import {
  useRegions
} from "./chunk-QZ6PT4QV.mjs";
import "./chunk-QL4XKIVL.mjs";
import "./chunk-4BGVSXZZ.mjs";
import "./chunk-FXYH54JP.mjs";
import "./chunk-774WSTCC.mjs";
import "./chunk-DEQUVHHE.mjs";
import "./chunk-QZ7TP4HQ.mjs";

// src/routes/orders/order-list/components/order-list-table/order-list-table.tsx
import { Container, Heading } from "@medusajs/ui";
import { keepPreviousData } from "@tanstack/react-query";
import { useTranslation as useTranslation3 } from "react-i18next";

// src/routes/orders/order-list/components/order-list-table/use-order-table-filters.tsx
import { useMemo } from "react";
import { useTranslation } from "react-i18next";
import { createDataTableFilterHelper } from "@medusajs/ui";
var filterHelper = createDataTableFilterHelper();
var useOrderTableFilters = () => {
  const { t } = useTranslation();
  const dateFilters = useDataTableDateFilters();
  const { regions } = useRegions({
    limit: 1e3,
    fields: "id,name"
  });
  const { sales_channels } = useSalesChannels({
    limit: 1e3,
    fields: "id,name"
  });
  return useMemo(() => {
    const filters = [...dateFilters];
    if (regions?.length) {
      filters.push(
        filterHelper.accessor("region_id", {
          label: t("fields.region"),
          type: "multiselect",
          options: regions.map((r) => ({
            label: r.name,
            value: r.id
          }))
        })
      );
    }
    if (sales_channels?.length) {
      filters.push(
        filterHelper.accessor("sales_channel_id", {
          label: t("fields.salesChannel"),
          type: "multiselect",
          options: sales_channels.map((s) => ({
            label: s.name,
            value: s.id
          }))
        })
      );
    }
    return filters;
  }, [regions, sales_channels, dateFilters, t]);
};

// src/routes/orders/order-list/components/order-list-table/configurable-order-list-table.tsx
import { useTranslation as useTranslation2 } from "react-i18next";

// src/routes/orders/order-list/components/order-list-table/order-table-adapter.tsx
function createOrderTableAdapter() {
  return createTableAdapter({
    entity: "orders",
    queryPrefix: "o",
    pageSize: 20,
    columnAdapter: orderColumnAdapter,
    useData: (fields, params) => {
      const { orders, count, isError, error, isLoading } = useOrders(
        {
          fields,
          ...params
        },
        {
          placeholderData: (previousData, previousQuery) => {
            const prevFields = previousQuery?.[previousQuery.length - 1]?.query?.fields;
            if (prevFields && prevFields !== fields) {
              return void 0;
            }
            return previousData;
          }
        }
      );
      return {
        data: orders,
        count,
        isLoading,
        isError,
        error
      };
    },
    getRowHref: (row) => `/orders/${row.id}`,
    emptyState: {
      empty: {
        heading: "No orders found"
      }
    }
  });
}
function useOrderTableAdapter() {
  const filters = useOrderTableFilters();
  const adapter = createOrderTableAdapter();
  return {
    ...adapter,
    filters
  };
}

// src/routes/orders/order-list/components/order-list-table/configurable-order-list-table.tsx
import { jsx } from "react/jsx-runtime";
var ConfigurableOrderListTable = () => {
  const { t } = useTranslation2();
  const orderAdapter = useOrderTableAdapter();
  return /* @__PURE__ */ jsx(
    ConfigurableDataTable,
    {
      adapter: orderAdapter,
      heading: t("orders.domain"),
      layout: "fill"
    }
  );
};

// src/routes/orders/order-list/const.ts
var DEFAULT_PROPERTIES = [
  "id",
  "status",
  "created_at",
  "email",
  "display_id",
  "payment_status",
  "fulfillment_status",
  "total",
  "currency_code"
];
var DEFAULT_RELATIONS = ["*customer", "*sales_channel"];
var DEFAULT_FIELDS = `${DEFAULT_PROPERTIES.join(
  ","
)},${DEFAULT_RELATIONS.join(",")}`;

// src/routes/orders/order-list/components/order-list-table/order-list-table.tsx
import { jsx as jsx2, jsxs } from "react/jsx-runtime";
var PAGE_SIZE = 20;
var OrderListTable = () => {
  const { t } = useTranslation3();
  const isViewConfigEnabled = useFeatureFlag("view_configurations");
  if (isViewConfigEnabled) {
    return /* @__PURE__ */ jsx2(ConfigurableOrderListTable, {});
  }
  const { searchParams, raw } = useOrderTableQuery({
    pageSize: PAGE_SIZE
  });
  const { orders, count, isError, error, isLoading } = useOrders(
    {
      fields: DEFAULT_FIELDS,
      ...searchParams
    },
    {
      placeholderData: keepPreviousData
    }
  );
  const filters = useOrderTableFilters();
  const columns = useOrderTableColumns({});
  const { table } = useDataTable({
    data: orders ?? [],
    columns,
    enablePagination: true,
    count,
    pageSize: PAGE_SIZE
  });
  if (isError) {
    throw error;
  }
  return /* @__PURE__ */ jsxs(Container, { className: "divide-y p-0", children: [
    /* @__PURE__ */ jsx2("div", { className: "flex items-center justify-between px-6 py-4", children: /* @__PURE__ */ jsx2(Heading, { children: t("orders.domain") }) }),
    /* @__PURE__ */ jsx2(
      _DataTable,
      {
        columns,
        table,
        pagination: true,
        navigateTo: (row) => `/orders/${row.original.id}`,
        filters,
        count,
        search: true,
        isLoading,
        pageSize: PAGE_SIZE,
        orderBy: [
          { key: "display_id", label: t("orders.fields.displayId") },
          { key: "created_at", label: t("fields.createdAt") },
          { key: "updated_at", label: t("fields.updatedAt") }
        ],
        queryObject: raw,
        noRecords: {
          message: t("orders.list.noRecordsMessage")
        }
      }
    )
  ] });
};

// src/routes/orders/order-list/order-list.tsx
import { jsx as jsx3 } from "react/jsx-runtime";
var OrderList = () => {
  const { getWidgets } = useExtension();
  return /* @__PURE__ */ jsx3(
    SingleColumnPage,
    {
      widgets: {
        after: getWidgets("order.list.after"),
        before: getWidgets("order.list.before")
      },
      hasOutlet: false,
      children: /* @__PURE__ */ jsx3(OrderListTable, {})
    }
  );
};
export {
  OrderList as Component
};
