// @mercurjs/b2c-core/.medusa/server/src/workflows/hooks/product-created.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSecondaryCategories = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const seller_1 = require("../../modules/seller");
const seller_shipping_profile_1 = __importDefault(require("../../links/seller-shipping-profile"));
const utils_2 = require("../attribute/utils");
const secondary_categories_1 = require("../../modules/secondary_categories");
const getVariantInventoryItemIds = async (variantId, container) => {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const items = await query.graph({
        entity: "product_variant",
        fields: ["inventory_items.inventory_item_id"],
        filters: {
            id: variantId,
        },
    });
    return items.data
        .map((item) => item.inventory_items.map((ii) => ii.inventory_item_id))
        .flat(2);
};
const assignDefaultSellerShippingProfile = async (container, product_id, seller_id) => {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const link = container.resolve(utils_1.ContainerRegistrationKeys.LINK);
    const { data: [existingLink], } = await query.graph({
        entity: "product_shipping_profile",
        fields: ["*"],
        filters: {
            product_id,
        },
    });
    if (existingLink) {
        return;
    }
    const { data: shippingProfiles } = await query.graph({
        entity: seller_shipping_profile_1.default.entryPoint,
        fields: ["shipping_profile.id", "shipping_profile.type"],
        filters: {
            seller_id,
        },
    });
    const [profile] = shippingProfiles.filter((relation) => relation.shipping_profile.type === "default");
    if (!profile) {
        return;
    }
    await link.create({
        [utils_1.Modules.PRODUCT]: {
            product_id,
        },
        [utils_1.Modules.FULFILLMENT]: {
            shipping_profile_id: profile.shipping_profile.id,
        },
    });
};
const getSecondaryCategories = async (secondaryCategoriesIds, container) => {
    const secondaryCategoryService = container.resolve(secondary_categories_1.SECONDARY_CATEGORY_MODULE);
    const existingCategories = await secondaryCategoryService.listSecondaryCategories({
        category_id: secondaryCategoriesIds,
    });
    const existingMap = new Map(existingCategories.map((cat) => [cat.id, cat]));
    const results = [];
    for (const id of secondaryCategoriesIds) {
        if (existingMap.has(id)) {
            results.push(existingMap.get(id));
        }
        else {
            const created = await secondaryCategoryService.createSecondaryCategories({
                category_id: id,
            });
            results.push(created);
        }
    }
    return results;
};
exports.getSecondaryCategories = getSecondaryCategories;
const createSecondaryCategories = async (products, additional_data, container) => {
    const links = [];
    products.map(async (product) => {
        if (additional_data?.secondary_categories?.length > 0) {
            const secondaryCategoriesIds = additional_data.secondary_categories.find((s) => s.handle === product.handle)?.secondary_categories_ids ?? [];
            const mappedSecondaryCategories = await (0, exports.getSecondaryCategories)(secondaryCategoriesIds, container);
            mappedSecondaryCategories.map((secondaryCategory) => {
                links.push({
                    [utils_1.Modules.PRODUCT]: {
                        product_id: product.id,
                    },
                    [secondary_categories_1.SECONDARY_CATEGORY_MODULE]: {
                        secondary_category_id: secondaryCategory.id,
                    },
                });
            });
            return links;
        }
    });
    return links;
};
core_flows_1.createProductsWorkflow.hooks.productsCreated(async ({ products, additional_data, }, { container }) => {
    await (0, utils_2.productsCreatedHookHandler)({
        products,
        additional_data,
        container,
    });
    const remoteLink = container.resolve(utils_1.ContainerRegistrationKeys.REMOTE_LINK);
    if (!additional_data?.seller_id) {
        return new workflows_sdk_1.StepResponse(undefined, null);
    }
    const variants = products.map((p) => p.variants).flat();
    const remoteLinks = products.map((product) => {
        return {
            [seller_1.SELLER_MODULE]: {
                seller_id: additional_data.seller_id,
            },
            [utils_1.Modules.PRODUCT]: {
                product_id: product.id,
            },
        };
    });
    for (const variant of variants) {
        if (variant.manage_inventory) {
            const inventoryItemIds = await getVariantInventoryItemIds(variant.id, container);
            inventoryItemIds.forEach((inventory_item_id) => {
                remoteLinks.push({
                    [seller_1.SELLER_MODULE]: {
                        seller_id: additional_data.seller_id,
                    },
                    [utils_1.Modules.INVENTORY]: {
                        inventory_item_id,
                    },
                });
            });
        }
    }
    const secondaryCategories = await createSecondaryCategories(products, additional_data, container);
    await Promise.all(products.map((p) => assignDefaultSellerShippingProfile(container, p.id, additional_data.seller_id)));
    await remoteLink.create([...remoteLinks, ...secondaryCategories]);
    await container.resolve(utils_1.Modules.EVENT_BUS).emit({
        name: framework_1.AlgoliaEvents.PRODUCTS_CHANGED,
        data: { ids: products.map((product) => product.id) },
    });
    return new workflows_sdk_1.StepResponse(undefined, products.map((product) => product.id));
}, async (productIds, { container }) => {
    if (!productIds) {
        return;
    }
    const remoteLink = container.resolve(utils_1.ContainerRegistrationKeys.REMOTE_LINK);
    await remoteLink.dismiss(productIds.map((productId) => ({
        [utils_1.Modules.PRODUCT]: {
            product_id: productId,
        },
    })));
});
