// @mercurjs/b2c-core/.medusa/server/src/workflows/hooks/product-updated.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateProductSubcategories = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const framework_1 = require("@mercurjs/framework");
const utils_2 = require("../attribute/utils");
const secondary_categories_1 = require("../../modules/secondary_categories");
const product_created_1 = require("./product-created");
const secondary_category_product_1 = __importDefault(require("../../links/secondary-category-product"));
const updateProductSubcategories = async (products, additional_data, container) => {
    const link = container.resolve(utils_1.ContainerRegistrationKeys.LINK);
    const secondaryCategoryService = container.resolve(secondary_categories_1.SECONDARY_CATEGORY_MODULE);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const entries = additional_data?.secondary_categories ?? [];
    if (!entries.length || !products?.length)
        return;
    const byId = new Map(entries.map((e) => [e.product_id, e]));
    await Promise.all(products.map(async (product) => {
        const match = byId.get(product.id);
        if (!match)
            return;
        const toAddRaw = match.add ?? match.secondary_categories_ids ?? [];
        const toRemoveRaw = match.remove ?? [];
        const toAddIds = [...new Set(toAddRaw.filter(Boolean))];
        const toRemoveIds = [...new Set(toRemoveRaw.filter(Boolean))];
        let confirmedAddIds = [];
        if (toAddIds.length) {
            const existingOrCreated = await (0, product_created_1.getSecondaryCategories)(toAddIds, container);
            confirmedAddIds = existingOrCreated.map((c) => c.id);
        }
        if (toRemoveIds.length) {
            await Promise.all(toRemoveIds.map(async (secId) => {
                const secondaryCategories = await secondaryCategoryService.listSecondaryCategories({
                    category_id: secId
                });
                const { data: [secondaryCategoryLink] } = await query.graph({
                    entity: secondary_category_product_1.default.entryPoint,
                    fields: ['secondary_category_id'],
                    filters: {
                        secondary_category_id: secondaryCategories.map((secondaryCategory) => secondaryCategory.id),
                        product_id: product.id
                    }
                });
                link
                    .dismiss({
                    [utils_1.Modules.PRODUCT]: { product_id: product.id },
                    [secondary_categories_1.SECONDARY_CATEGORY_MODULE]: {
                        secondary_category_id: secondaryCategoryLink.secondary_category_id
                    }
                })
                    .catch(() => { });
            }));
        }
        if (confirmedAddIds.length) {
            await Promise.all(confirmedAddIds.map((secId) => link
                .create({
                [utils_1.Modules.PRODUCT]: { product_id: product.id },
                [secondary_categories_1.SECONDARY_CATEGORY_MODULE]: { secondary_category_id: secId },
            })
                .catch(() => { })));
        }
    }));
};
exports.updateProductSubcategories = updateProductSubcategories;
core_flows_1.updateProductsWorkflow.hooks.productsUpdated(async ({ products, additional_data }, { container }) => {
    await (0, utils_2.productsUpdatedHookHandler)({
        products,
        additional_data,
        container,
    });
    await (0, exports.updateProductSubcategories)(products, additional_data, container);
    await container.resolve(utils_1.Modules.EVENT_BUS).emit({
        name: framework_1.AlgoliaEvents.PRODUCTS_CHANGED,
        data: { ids: products.map((product) => product.id) },
    });
});
