// @mercurjs/requests/.medusa/server/src/workflows/order-return-request/workflows/update-return-request.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateOrderReturnRequestWorkflow = void 0;
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const steps_1 = require("../steps");
const proceed_return_request_1 = require("./proceed-return-request");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const return_request_order_1 = __importDefault(require("../../../links/return-request-order"));
const utils_1 = require("@medusajs/framework/utils");
exports.updateOrderReturnRequestWorkflow = (0, workflows_sdk_1.createWorkflow)("update-order-return-request", function (input) {
    (0, workflows_sdk_1.when)(input, (input) => input.status === "refunded").then(() => {
        proceed_return_request_1.proceedReturnRequestWorkflow.runAsStep({ input });
    });
    const request = (0, steps_1.updateOrderReturnRequestStep)(input);
    const requestId = (0, workflows_sdk_1.transform)(request, (request) => request.id);
    const order = (0, core_flows_1.useQueryGraphStep)({
        entity: return_request_order_1.default.entryPoint,
        fields: ["order.returns.id", "order_return_request.seller.id"],
        filters: {
            order_return_request_id: requestId,
        },
    });
    const links = (0, workflows_sdk_1.transform)(order, (order) => {
        const returns = order.data[0].order.returns;
        const toLink = Array.isArray(returns) ? returns : [returns];
        const seller = order.data[0].order_return_request.seller.id;
        return toLink.map((r) => {
            return {
                [framework_1.SELLER_MODULE]: {
                    seller_id: seller,
                },
                [utils_1.Modules.ORDER]: {
                    return_id: r.id,
                },
            };
        });
    });
    (0, core_flows_1.createRemoteLinkStep)(links);
    const orderReturnRequestUpdatedHook = (0, workflows_sdk_1.createHook)("orderReturnRequestUpdated", {
        requestId,
    });
    return new workflows_sdk_1.WorkflowResponse(request, {
        hooks: [orderReturnRequestUpdatedHook],
    });
});
