// @mercurjs/b2c-core/.medusa/server/src/api/vendor/product-categories/query-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorProductCategoryProductsQueryConfig = exports.vendorProductCategoryProductsFields = exports.vendorProductCategoryQueryConfig = exports.vendorProductCategoryFields = void 0;
exports.vendorProductCategoryFields = [
    'id',
    'name',
    'description',
    'handle',
    'rank',
    'parent_category_id',
    'created_at',
    'updated_at',
    'metadata',
    '*parent_category',
    '*category_children'
];
exports.vendorProductCategoryQueryConfig = {
    list: {
        defaults: exports.vendorProductCategoryFields,
        isList: true
    },
    retrieve: {
        defaults: exports.vendorProductCategoryFields,
        isList: false
    }
};
exports.vendorProductCategoryProductsFields = [
    'id',
    'title',
    'status',
    'thumbnail',
    'handle',
    'collection.id',
    'collection.title',
    'variants.id'
];
exports.vendorProductCategoryProductsQueryConfig = {
    list: {
        defaults: exports.vendorProductCategoryProductsFields,
        isList: true
    },
    retrieve: {
        defaults: exports.vendorProductCategoryProductsFields,
        isList: false
    }
};
