// @mercurjs/commission/.medusa/server/src/workflows/commission/steps/calculate-commission-lines.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.calculateCommissionLinesStep = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const commission_1 = require("../../../modules/commission");
async function calculateFlatCommission(rate, currency, container) {
    const priceService = container.resolve(utils_1.Modules.PRICING);
    const priceSet = await priceService.retrievePriceSet(rate.price_set_id, {
        relations: ["prices"],
    });
    const price = priceSet.prices?.find((p) => p.currency_code === currency);
    return price?.amount || utils_1.MathBN.convert(0);
}
async function calculatePercentageCommission(rate, item, currency, container) {
    const total = utils_1.MathBN.convert(item.total);
    const taxValue = utils_1.MathBN.convert(item.tax_total);
    const calculationBase = rate.include_tax ? total : total.minus(taxValue);
    const commissionValue = utils_1.MathBN.mult(calculationBase, utils_1.MathBN.div(rate.percentage_rate, 100));
    const priceService = container.resolve(utils_1.Modules.PRICING);
    const minPriceSet = rate.min_price_set_id
        ? await priceService.retrievePriceSet(rate.min_price_set_id, {
            relations: ["prices"],
        })
        : undefined;
    const maxPriceSet = rate.max_price_set_id
        ? await priceService.retrievePriceSet(rate.max_price_set_id, {
            relations: ["prices"],
        })
        : undefined;
    const minValue = minPriceSet?.prices?.find((p) => p.currency_code === currency)
        ?.amount || utils_1.MathBN.convert(0);
    const maxValue = maxPriceSet?.prices?.find((p) => p.currency_code === currency)
        ?.amount || utils_1.MathBN.convert(Number.POSITIVE_INFINITY);
    return utils_1.MathBN.max(minValue, utils_1.MathBN.min(maxValue, commissionValue));
}
async function calculateCommissionValue(rate, item, currency, container) {
    if (rate.type === "flat") {
        return calculateFlatCommission(rate, currency, container);
    }
    if (rate.type === "percentage") {
        return calculatePercentageCommission(rate, item, currency, container);
    }
    return utils_1.MathBN.convert(0);
}
exports.calculateCommissionLinesStep = (0, workflows_sdk_1.createStep)("calculate-commission-lines", async ({ order_id, seller_id }, { container }) => {
    const orderService = container.resolve(utils_1.Modules.ORDER);
    const order = await orderService.retrieveOrder(order_id, {
        relations: ["items"],
        // At least one of the computed totals fields should be requested in select,
        // in order for decorateTotals to be called
        select: ["*", "item_total"],
    });
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const commissionService = container.resolve(commission_1.COMMISSION_MODULE);
    const commissionLines = [];
    for (const item of order.items) {
        const { data: [product], } = await query.graph({
            entity: "product",
            fields: ["categories.id"],
            filters: {
                id: item.product_id,
            },
        });
        const commissionRule = await commissionService.selectCommissionForProductLine({
            product_category_id: product.categories[0]?.id || "",
            product_type_id: item.product_type_id || "",
            seller_id: seller_id,
        });
        if (commissionRule) {
            commissionLines.push({
                item_line_id: item.id,
                value: await calculateCommissionValue(commissionRule.rate, item, order.currency_code, container),
                currency_code: order.currency_code,
                rule_id: commissionRule.id,
            });
        }
    }
    return new workflows_sdk_1.StepResponse(commissionLines);
});
