// @mercurjs/commission/.medusa/server/src/workflows/commission/steps/list-commission-lines.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listCommissionLinesStep = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const commission_1 = require("../../../modules/commission");
const framework_1 = require("@mercurjs/framework");
exports.listCommissionLinesStep = (0, workflows_sdk_1.createStep)("list-commission-lines", async (input, { container }) => {
    const { pagination, filters } = input;
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const service = container.resolve(commission_1.COMMISSION_MODULE);
    const createdFilter = {};
    const itemLineIdFilter = {};
    if (filters.start_date) {
        createdFilter.$gte = filters.start_date;
    }
    if (filters.end_date) {
        createdFilter.$lte = filters.end_date;
    }
    if (filters.seller_id) {
        const { data: sellerOrders } = await query.graph({
            entity: framework_1.SELLER_ORDER_LINK,
            fields: ["*", "order.items.id"],
            filters: {
                seller_id: filters.seller_id,
                created_at: createdFilter,
            },
        });
        itemLineIdFilter.$in = sellerOrders
            .flatMap((o) => o.order.items)
            .map((i) => i.id);
    }
    const [commissionLines, count] = await service.listAndCountCommissionLines({
        item_line_id: itemLineIdFilter,
        created_at: createdFilter,
    }, {
        take: pagination.take,
        skip: pagination.skip,
    });
    if (input.expand) {
        const itemIds = commissionLines.map((line) => line.item_line_id);
        const ruleIds = commissionLines.map((line) => line.rule_id);
        const { data: rules } = await query.graph({
            entity: "commission_rule",
            fields: ["*", "rate.*"],
            filters: {
                id: ruleIds,
            },
        });
        const { data: orders } = await query.graph({
            entity: "order",
            fields: ["*", "seller.id", "seller.name", "items.id"],
            filters: {
                items: {
                    id: itemIds,
                },
            },
        });
        const expandedLines = commissionLines.map((line) => {
            const order = orders.find((o) => o.items.some((i) => i.id === line.item_line_id));
            const rule = rules.find((r) => r.id === line.rule_id);
            return {
                ...line,
                order,
                rule: rule ?? null
            };
        });
        return new workflows_sdk_1.StepResponse({ lines: expandedLines, count });
    }
    return new workflows_sdk_1.StepResponse({ lines: commissionLines, count });
});
