// @mercurjs/requests/.medusa/server/src/api/vendor/requests/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorRequestsMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const seller_request_1 = __importDefault(require("../../../links/seller-request"));
const framework_2 = require("@mercurjs/framework");
const apply_request_status_filter_1 = require("../../middlewares/apply-request-status-filter");
const apply_request_type_filter_1 = require("../../middlewares/apply-request-type-filter");
const query_config_1 = require("./query-config");
const validators_1 = require("./validators");
exports.vendorRequestsMiddlewares = [
    {
        method: ["GET"],
        matcher: "/vendor/requests",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetRequestsParams, query_config_1.vendorRequestsConfig.list),
            (0, framework_2.filterBySellerId)(),
            (0, apply_request_status_filter_1.applyRequestsStatusFilter)(),
            (0, apply_request_type_filter_1.applyRequestsTypeFilter)(),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/requests/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetRequestsParams, query_config_1.vendorRequestsConfig.retrieve),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_request_1.default.entryPoint,
                filterField: "request_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/requests/:id",
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_1.VendorUpdateRequestData),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetRequestsParams, query_config_1.vendorRequestsConfig.retrieve),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_request_1.default.entryPoint,
                filterField: "request_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/requests",
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_1.VendorCreateRequest),
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetRequestsParams, query_config_1.vendorRequestsConfig.retrieve),
        ],
    },
];
