// @mercurjs/requests/.medusa/server/src/api/vendor/requests/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorUpdateRequestData = exports.VendorCreateRequest = exports.VendorGetRequestsParams = void 0;
const zod_1 = require("zod");
const validators_1 = require("@medusajs/medusa/api/utils/validators");
exports.VendorGetRequestsParams = (0, validators_1.createFindParams)({
    offset: 0,
    limit: 50
}).extend({
    type: zod_1.z
        .enum([
        'product_collection',
        'product_collection_update',
        'product_category',
        'product',
        'review_remove',
        'product_type',
        'product_tag'
    ])
        .optional(),
    status: zod_1.z.enum(['accepted', 'rejected', 'pending']).optional()
});
/**
 * @schema ProductCategoryRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [product_category]
 *   data:
 *     type: object
 *     properties:
 *       name:
 *         type: string
 *         description: The name of the product category
 *       handle:
 *         type: string
 *         description: The description of the product category
 *       description:
 *         type: string
 *         description: The description of the product category
 *       parent_category_id:
 *         type: string
 *         description: The id of the parent category
 */
const ProductCategoryRequest = zod_1.z.object({
    type: zod_1.z.literal('product_category'),
    data: zod_1.z.object({
        name: zod_1.z.string(),
        handle: zod_1.z.string(),
        description: zod_1.z.string().optional(),
        parent_category_id: zod_1.z.string().nullable().default(null)
    })
});
/**
 * @schema ProductCollectionRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [product_collection]
 *   data:
 *     type: object
 *     properties:
 *       title:
 *         type: string
 *         description: The title of the product collection
 *       handle:
 *         type: string
 *         description: The description of the product category
 */
const ProductCollectionRequest = zod_1.z.object({
    type: zod_1.z.literal('product_collection'),
    data: zod_1.z.object({
        title: zod_1.z.string(),
        handle: zod_1.z.string()
    })
});
/**
 * @schema ProductCollectionUpdateRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [product_collection_update]
 *   data:
 *     type: object
 *     required:
 *       - collection_id
 *     properties:
 *       collection_id:
 *         type: string
 *         description: The ID of the existing collection to update
 *       title:
 *         type: string
 *         description: The new title of the product collection
 *       handle:
 *         type: string
 *         description: The new handle of the product collection
 */
const ProductCollectionUpdateRequest = zod_1.z.object({
    type: zod_1.z.literal('product_collection_update'),
    data: zod_1.z.object({
        collection_id: zod_1.z.string(),
        title: zod_1.z.string().optional(),
        handle: zod_1.z.string().optional()
    })
});
/**
 * @schema ReviewRemoveRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [review_remove]
 *   data:
 *     type: object
 *     properties:
 *       review_id:
 *         type: string
 *         description: Id of the review to remove
 *       reason:
 *         type: string
 *         description: The reason to remove review
 */
const ReviewRemoveRequest = zod_1.z.object({
    type: zod_1.z.literal('review_remove'),
    data: zod_1.z.object({
        review_id: zod_1.z.string(),
        reason: zod_1.z.string()
    })
});
/**
 * @schema ProductTypeRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [product_type]
 *   data:
 *     type: object
 *     properties:
 *       value:
 *         type: string
 *         description: The product type value
 *       metadata:
 *         type: object
 *         description: The product type metadata
 */
const ProductTypeRequest = zod_1.z.object({
    type: zod_1.z.literal('product_type'),
    data: zod_1.z.object({
        value: zod_1.z.string(),
        metadata: zod_1.z.record(zod_1.z.unknown()).nullish()
    })
});
/**
 * @schema ProductTagRequest
 * type: object
 * required:
 *   - type
 *   - data
 * properties:
 *   type:
 *     type: string
 *     description: The type of the request
 *     enum: [product_tag]
 *   data:
 *     type: object
 *     properties:
 *       value:
 *         type: string
 *         description: The product tag value
 *       metadata:
 *         type: object
 *         description: The product tag metadata
 */
const ProductTagRequest = zod_1.z.object({
    type: zod_1.z.literal('product_tag'),
    data: zod_1.z.object({
        value: zod_1.z.string(),
        metadata: zod_1.z.record(zod_1.z.unknown()).nullish()
    })
});
exports.VendorCreateRequest = zod_1.z.object({
    request: zod_1.z.discriminatedUnion('type', [
        ProductCategoryRequest,
        ProductCollectionRequest,
        ProductCollectionUpdateRequest,
        ReviewRemoveRequest,
        ProductTypeRequest,
        ProductTagRequest
    ])
});
const UpdateProductCollectionRequest = zod_1.z.object({
    type: zod_1.z.literal('product_collection'),
    data: zod_1.z.object({
        title: zod_1.z.string().optional(),
        handle: zod_1.z.string().optional()
    })
});
const UpdateReviewRemoveRequest = zod_1.z.object({
    type: zod_1.z.literal('review_remove'),
    data: zod_1.z.object({
        review_id: zod_1.z.string().optional(),
        reason: zod_1.z.string().optional()
    })
});
const UpdateProductCategoryRequest = zod_1.z.object({
    type: zod_1.z.literal('product_category'),
    data: zod_1.z.object({
        name: zod_1.z.string().optional(),
        handle: zod_1.z.string().optional(),
        description: zod_1.z.string().optional(),
        parent_category_id: zod_1.z.string().nullable().optional()
    })
});
exports.VendorUpdateRequestData = zod_1.z.object({
    request: zod_1.z.discriminatedUnion('type', [
        UpdateProductCategoryRequest,
        UpdateProductCollectionRequest,
        UpdateReviewRemoveRequest,
        ProductTypeRequest,
        ProductTagRequest
    ])
});
