// @mercurjs/b2c-core/.medusa/server/src/api/vendor/inventory-items/[id]/location-levels/[location_id]/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DELETE = exports.GET = exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const framework_1 = require("@mercurjs/framework");
const helpers_1 = require("@medusajs/medusa/api/admin/inventory-items/helpers");
/**
 * @oas [post] /vendor/inventory-items/{id}/location-levels/{location_id}
 * operationId: "VendorUpdateInventoryLevel"
 * summary: "Update inventory level"
 * description: "Updates inventory level of the InventoryItem in the specified location"
 * x-authenticated: true
 * parameters:
 *   - in: path
 *     name: id
 *     required: true
 *     description: The ID of the InventoryItem.
 *     schema:
 *       type: string
 *   - in: path
 *     name: location_id
 *     required: true
 *     description: The ID of the Stock Location.
 *     schema:
 *       type: string
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorUpdateInventoryLevel"
 * responses:
 *   "202":
 *     description: Accepted
 * tags:
 *   - Vendor Inventory Items
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    await core_flows_1.updateInventoryLevelsWorkflow.run({
        input: {
            updates: [
                {
                    location_id: req.params.location_id,
                    inventory_item_id: req.params.id,
                    ...req.validatedBody
                }
            ]
        },
        container: req.scope
    });
    const eventBus = req.scope.resolve(utils_1.Modules.EVENT_BUS);
    await eventBus.emit({
        name: framework_1.IntermediateEvents.INVENTORY_ITEM_CHANGED,
        data: { id: req.params.id }
    });
    const { data: [location_level] } = await query.graph({
        entity: 'inventory_level',
        fields: req.queryConfig.fields,
        filters: {
            inventory_item_id: req.params.id,
            location_id: req.params.location_id
        }
    });
    res.json({
        location_level
    });
};
exports.POST = POST;
/**
 * @oas [get] /vendor/inventory-items/{id}/location-levels/{location_id}
 * operationId: "VendorGetInventoryLevel"
 * summary: "Get inventory level"
 * description: "Retrieves inventory level of the InventoryItem in the specified location"
 * x-authenticated: true
 * parameters:
 *   - in: path
 *     name: id
 *     required: true
 *     description: The ID of the InventoryItem.
 *     schema:
 *       type: string
 *   - in: path
 *     name: location_id
 *     required: true
 *     description: The ID of the Stock Location.
 *     schema:
 *       type: string
 * responses:
 *   "200":
 *     description: Inventory level
 * tags:
 *   - Vendor Inventory Items
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const GET = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [location_level] } = await query.graph({
        entity: 'inventory_level',
        fields: req.queryConfig.fields,
        filters: {
            inventory_item_id: req.params.id,
            location_id: req.params.location_id
        }
    });
    res.json({
        location_level
    });
};
exports.GET = GET;
/**
 * @oas [delete] /vendor/inventory-items/{id}/location-levels/{location_id}
 * operationId: "VendorDeleteInventoryLevel"
 * summary: "Delete inventory level"
 * description: "Deletes inventory level of the InventoryItem in the specified location"
 * x-authenticated: true
 * parameters:
 *   - in: path
 *     name: id
 *     required: true
 *     description: The ID of the InventoryItem.
 *     schema:
 *       type: string
 *   - in: path
 *     name: location_id
 *     required: true
 *     description: The ID of the Stock Location.
 *     schema:
 *       type: string
 * responses:
 *   "200":
 *     description: Inventory level
 * tags:
 *   - Vendor Inventory Items
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const DELETE = async (req, res) => {
    const { id, location_id } = req.params;
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const result = await query.graph({
        entity: "inventory_level",
        filters: { inventory_item_id: id, location_id },
        fields: ["id", "reserved_quantity"],
    });
    if (!result.data.length) {
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.NOT_FOUND, `Inventory Level for Item ${id} at Location ${location_id} not found`);
    }
    const { id: levelId, reserved_quantity: reservedQuantity } = result.data[0];
    if (reservedQuantity > 0) {
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.NOT_ALLOWED, `Cannot remove Inventory Level ${id} at Location ${location_id} because there are reservations at location`);
    }
    const deleteInventoryLevelWorkflow = (0, core_flows_1.deleteInventoryLevelsWorkflow)(req.scope);
    await deleteInventoryLevelWorkflow.run({
        input: {
            id: [levelId],
        },
    });
    const inventoryItem = await (0, helpers_1.refetchInventoryItem)(id, req.scope, req.queryConfig.fields);
    res.status(200).json({
        id: levelId,
        object: "inventory-level",
        deleted: true,
        parent: inventoryItem,
    });
};
exports.DELETE = DELETE;
