// @mercurjs/b2c-core/.medusa/server/src/workflows/seller/workflows/import-seller-products.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.importSellerProductsWorkflow = void 0;
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_1 = require("@medusajs/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const steps_1 = require("../steps");
exports.importSellerProductsWorkflow = (0, workflows_sdk_1.createWorkflow)("import-seller-products", function (input) {
    const products = (0, core_flows_1.parseProductCsvStep)(input.file_content);
    const batchCreate = (0, steps_1.validateProductsToImportStep)(products);
    const created = core_flows_1.createProductsWorkflow.runAsStep({
        input: {
            products: batchCreate,
            additional_data: { seller_id: input.seller_id },
        },
    });
    const requestsPayload = (0, workflows_sdk_1.transform)({ created, input }, ({ created, input }) => {
        return created.map((p) => ({
            data: {
                ...p,
                product_id: p.id,
            },
            submitter_id: input.submitter_id,
            type: "product",
            status: "pending",
        }));
    });
    const eventPayload = (0, workflows_sdk_1.transform)({ requestsPayload, input }, ({ requestsPayload, input }) => ({
        request_payloads: requestsPayload,
        seller_id: input.seller_id,
        submitter_id: input.submitter_id,
    }));
    (0, core_flows_1.emitEventStep)({
        eventName: framework_1.ImportSellerProductsRequestUpdatedEvent.TO_CREATE,
        data: eventPayload,
    });
    return new workflows_sdk_1.WorkflowResponse(created);
});
