// @mercurjs/b2c-core/.medusa/server/src/workflows/cart/workflows/split-and-complete-cart.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitAndCompleteCartWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_2 = require("@medusajs/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const marketplace_1 = require("../../../modules/marketplace");
const seller_1 = require("../../../modules/seller");
const steps_1 = require("../../promotions/steps");
const steps_2 = require("../../split-order-payment/steps");
const steps_3 = require("../steps");
const utils_2 = require("../utils");
exports.splitAndCompleteCartWorkflow = (0, workflows_sdk_2.createWorkflow)({
    name: "split-and-complete-cart",
    idempotent: true,
}, function (input) {
    const existingOrderSet = (0, core_flows_1.useRemoteQueryStep)({
        entry_point: "order_set",
        fields: ["id", "cart_id"],
        variables: {
            filters: {
                cart_id: input.id,
            },
        },
        list: false,
    }).config({ name: "order-set-query" });
    const orderSet = (0, workflows_sdk_1.when)({ existingOrderSet }, ({ existingOrderSet }) => {
        return !existingOrderSet;
    }).then(() => {
        const cart = (0, core_flows_1.useRemoteQueryStep)({
            entry_point: "cart",
            fields: utils_2.completeCartFields,
            variables: {
                id: input.id,
            },
            list: false,
        }).config({ name: "cart-query" });
        (0, steps_3.validateCartSellersStep)((0, workflows_sdk_1.transform)({ cart }, ({ cart }) => ({
            line_items: cart.items,
        })));
        const validateCartShippingOptionsInput = (0, workflows_sdk_1.transform)({ cart }, ({ cart }) => ({
            cart_id: cart.id,
            option_ids: cart.shipping_methods.map((method) => method.shipping_option_id),
        }));
        const { sellerProducts, sellerShippingOptions } = (0, steps_3.validateCartShippingOptionsStep)(validateCartShippingOptionsInput);
        const paymentSessions = (0, core_flows_1.validateCartPaymentsStep)({ cart });
        const payment = (0, core_flows_1.authorizePaymentSessionStep)({
            id: paymentSessions[0].id,
            context: { cart_id: cart.id },
        });
        const { ordersToCreate, sellers, variants } = (0, workflows_sdk_1.transform)({ cart, sellerProducts, sellerShippingOptions }, ({ cart, sellerProducts, sellerShippingOptions }) => {
            const productSellerMap = new Map(sellerProducts.map((sp) => [sp.product_id, sp.seller_id]));
            const shippingOptionSellerMap = new Map(sellerShippingOptions.map((sp) => [
                sp.shipping_option_id,
                sp.seller_id,
            ]));
            const sellerLineItemsMap = new Map();
            const sellerShippingMethodsMap = new Map();
            const variantsMap = new Map();
            cart.items.forEach((item) => {
                const sellerId = productSellerMap.get(item.variant.product_id);
                const lineItems = sellerLineItemsMap.get(sellerId) || [];
                lineItems.push(item);
                sellerLineItemsMap.set(sellerId, lineItems);
                variantsMap.set(item.variant.id, item.variant);
            });
            cart.shipping_methods.forEach((method) => {
                const sellerId = shippingOptionSellerMap.get(method.shipping_option_id);
                sellerShippingMethodsMap.set(sellerId, method);
            });
            const sellers = Array.from(sellerLineItemsMap.keys());
            const ordersToCreate = sellers.map((sellerId) => {
                const sm = sellerShippingMethodsMap.get(sellerId);
                if (!sm) {
                    throw new utils_1.MedusaError(utils_1.MedusaError.Types.INVALID_DATA, "Seller shipping method not found!");
                }
                const items = sellerLineItemsMap.get(sellerId).map((item) => (0, utils_2.prepareLineItemData)({
                    item,
                    variant: item.variant,
                    unitPrice: item.unit_price,
                    compareAtUnitPrice: item.compare_at_unit_price,
                    isTaxInclusive: item.is_tax_inclusive,
                    quantity: item.quantity,
                    metadata: item?.metadata,
                    taxLines: item.tax_lines ?? [],
                    adjustments: item.adjustments ?? [],
                }));
                const itemAdjustments = items
                    .map((item) => item.adjustments ?? [])
                    .flat(1);
                const promoCodes = [...itemAdjustments]
                    .map((adjustment) => adjustment.code)
                    .filter(Boolean);
                return {
                    region_id: cart.region?.id,
                    customer_id: cart.customer?.id,
                    sales_channel_id: cart.sales_channel_id,
                    status: utils_1.OrderStatus.PENDING,
                    email: cart.email,
                    currency_code: cart.currency_code,
                    shipping_address: cart.shipping_address,
                    billing_address: cart.billing_address,
                    no_notification: false,
                    items,
                    promo_codes: promoCodes,
                    shipping_methods: [
                        {
                            name: sm.name,
                            description: sm.description,
                            amount: sm.amount,
                            is_tax_inclusive: sm.is_tax_inclusive,
                            shipping_option_id: sm.shipping_option_id,
                            data: sm.data,
                            metadata: sm.metadata,
                            tax_lines: (0, utils_2.prepareTaxLinesData)(sm.tax_lines ?? []),
                        },
                    ],
                };
            });
            return {
                ordersToCreate,
                sellers,
                variants: Array.from(variantsMap.values()),
            };
        });
        const promotionUsage = (0, workflows_sdk_1.transform)({ cart }, ({ cart }) => {
            const promotionUsage = [];
            const itemAdjustments = (cart.items ?? [])
                .map((item) => item.adjustments ?? [])
                .flat(1);
            const shippingAdjustments = (cart.shipping_methods ?? [])
                .map((item) => item.adjustments ?? [])
                .flat(1);
            for (const adjustment of itemAdjustments) {
                promotionUsage.push({
                    amount: adjustment.amount,
                    code: adjustment.code,
                });
            }
            for (const adjustment of shippingAdjustments) {
                promotionUsage.push({
                    amount: adjustment.amount,
                    code: adjustment.code,
                });
            }
            return { computedActions: promotionUsage, registrationContext: { customer_id: cart.customer?.id ?? null, customer_email: cart.email ?? null } };
        });
        (0, steps_1.registerUsageStep)(promotionUsage);
        const orderSet = (0, steps_3.createOrderSetStep)({
            cart_id: cart.id,
            customer_id: cart.customer_id,
            sales_channel_id: cart.sales_channel_id,
            payment_collection_id: payment.payment_collection_id,
        });
        const createdOrders = (0, core_flows_1.createOrdersStep)(ordersToCreate);
        const splitPaymentsToCreate = (0, workflows_sdk_1.transform)({ createdOrders, payment }, ({ createdOrders, payment }) => {
            return createdOrders.map((order) => ({
                order_id: order.id,
                status: "pending",
                currency_code: order.currency_code,
                authorized_amount: utils_1.MathBN.convert(
                // @ts-ignore
                order.summary?.accounting_total || 0).toNumber(),
                payment_collection_id: payment.payment_collection_id,
            }));
        });
        const reservationItemsData = (0, workflows_sdk_1.transform)({ createdOrders }, ({ createdOrders }) => createdOrders.reduce((acc, order) => {
            acc.push(...order.items.map((i) => ({
                variant_id: i.variant_id,
                quantity: i.quantity,
                id: i.id,
            })));
            return acc;
        }, []));
        const updateCartInput = (0, workflows_sdk_1.transform)({ cart }, ({ cart }) => ({
            id: cart.id,
            completed_at: new Date(),
        }));
        const formatedInventoryItems = (0, workflows_sdk_1.transform)({
            input: {
                sales_channel_id: cart.sales_channel_id,
                variants,
                items: reservationItemsData,
            },
        }, utils_2.prepareConfirmInventoryInput);
        const links = (0, workflows_sdk_1.transform)({
            createdOrders,
            sellers,
            orderSet,
            cart,
        }, ({ createdOrders, sellers, orderSet, cart }) => {
            const sellerOrderLinks = createdOrders.map((order, index) => ({
                [seller_1.SELLER_MODULE]: {
                    seller_id: sellers[index],
                },
                [utils_1.Modules.ORDER]: {
                    order_id: order.id,
                },
            }));
            const orderSetOrderLinks = createdOrders.map((order) => ({
                [marketplace_1.MARKETPLACE_MODULE]: {
                    order_set_id: orderSet.id,
                },
                [utils_1.Modules.ORDER]: {
                    order_id: order.id,
                },
            }));
            const orderPaymentLinks = createdOrders.map((order) => ({
                [utils_1.Modules.ORDER]: {
                    order_id: order.id,
                },
                [utils_1.Modules.PAYMENT]: {
                    payment_collection_id: cart.payment_collection.id,
                },
            }));
            return [
                ...sellerOrderLinks,
                ...orderSetOrderLinks,
                ...orderPaymentLinks,
            ];
        });
        const orderEvents = (0, workflows_sdk_1.transform)({ createdOrders }, ({ createdOrders }) => ({
            eventName: utils_1.OrderWorkflowEvents.PLACED,
            data: createdOrders.map((order) => ({
                id: order.id,
            })),
        }));
        (0, workflows_sdk_1.parallelize)((0, steps_2.createSplitOrderPaymentsStep)(splitPaymentsToCreate), (0, core_flows_1.createRemoteLinkStep)(links), (0, core_flows_1.reserveInventoryStep)(formatedInventoryItems), (0, core_flows_1.updateCartsStep)([updateCartInput]), (0, core_flows_1.emitEventStep)(orderEvents), (0, core_flows_1.emitEventStep)({
            eventName: framework_1.OrderSetWorkflowEvents.PLACED,
            data: {
                id: orderSet.id,
            },
        }).config({ name: "order-set-event" }));
        return orderSet;
    });
    const orderSetId = (0, workflows_sdk_1.transform)({ orderSet, existingOrderSet }, ({ orderSet, existingOrderSet }) => orderSet ? orderSet.id : existingOrderSet.id);
    const orderSetCreatedHook = (0, workflows_sdk_2.createHook)("orderSetCreated", {
        orderSetId,
    });
    return new workflows_sdk_2.WorkflowResponse({ id: orderSetId }, {
        hooks: [orderSetCreatedHook],
    });
});
