// @mercurjs/b2c-core/.medusa/server/src/api/vendor/stock-locations/[id]/fulfillment-sets/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const utils_2 = require("../../../../../shared/infra/http/utils");
const fulfillment_set_1 = require("../../../../../workflows/fulfillment-set");
/**
 * @oas [post] /vendor/stock-locations/{id}/fulfillment-sets
 * operationId: "VendorCreateStockLocationFulfillmentSet"
 * summary: "Create a Fulfillment Set"
 * description: "Creates a Fulfillment Set for a Stock Location."
 * x-authenticated: true
 * parameters:
 *   - in: path
 *     name: id
 *     required: true
 *     description: The ID of the Stock Location.
 *     schema:
 *       type: string
 *   - in: query
 *     name: fields
 *     description: The comma-separated fields to include in the response
 *     schema:
 *       type: string
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateStockLocationFulfillmentSet"
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             stock_location:
 *               $ref: "#/components/schemas/VendorStockLocation"
 * tags:
 *   - Vendor Stock Locations
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const seller = await (0, utils_2.fetchSellerByAuthActorId)(req.auth_context.actor_id, req.scope);
    await (0, fulfillment_set_1.createLocationFulfillmentSetAndAssociateWithSellerWorkflow)(req.scope).run({
        input: {
            location_id: req.params.id,
            fulfillment_set_data: {
                name: req.validatedBody.name,
                type: req.validatedBody.type,
            },
            seller_id: seller.id,
        },
    });
    const eventBus = req.scope.resolve(utils_1.Modules.EVENT_BUS);
    await eventBus.emit({
        name: framework_1.IntermediateEvents.STOCK_LOCATION_CHANGED,
        data: { id: req.params.id },
    });
    const { data: [stockLocation], } = await query.graph({
        entity: "stock_location",
        fields: req.queryConfig.fields,
        filters: {
            id: req.params.id,
        },
    }, { throwIfKeyNotFound: true });
    res.status(200).json({ stock_location: stockLocation });
};
exports.POST = POST;
