// @mercurjs/resend/.medusa/server/src/providers/resend/types/templates.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResendNotificationTemplates = void 0;
var ResendNotificationTemplates;
(function (ResendNotificationTemplates) {
    ResendNotificationTemplates["BUYER_ACCOUNT_CREATED"] = "buyerAccountCreatedEmailTemplate";
    ResendNotificationTemplates["BUYER_NEW_ORDER"] = "buyerNewOrderEmailTemplate";
    ResendNotificationTemplates["BUYER_CANCELED_ORDER"] = "buyerCancelOrderEmailTemplate";
    ResendNotificationTemplates["BUYER_ORDER_SHIPPED"] = "buyerOrderShippedEmailTemplate";
    ResendNotificationTemplates["SELLER_ACCOUNT_UPDATES_SUBMISSION"] = "sellerAccountSubmissionEmailTemplate";
    ResendNotificationTemplates["SELLER_ACCOUNT_UPDATES_APPROVAL"] = "sellerAccountApprovedEmailTemplate";
    ResendNotificationTemplates["SELLER_ACCOUNT_UPDATES_REJECTION"] = "sellerAccountRejectedEmailTemplate";
    ResendNotificationTemplates["SELLER_NEW_ORDER"] = "sellerNewOrderEmailTemplate";
    ResendNotificationTemplates["SELLER_CANCELED_ORDER"] = "sellerCanceledOrderEmailTemplate";
    ResendNotificationTemplates["SELLER_ORDER_SHIPPED"] = "sellerOrderShippingEmailTemplate";
    ResendNotificationTemplates["SELLER_TEAM_MEMBER_INVITATION"] = "sellerTeamInviteEmailTemplate";
    ResendNotificationTemplates["SELLER_VERIFY_EMAIL_TEMPLATE"] = "sellerVerifyEmailTemplate";
    ResendNotificationTemplates["FORGOT_PASSWORD"] = "forgotPasswordEmailTemplate";
    ResendNotificationTemplates["NEW_SELLER_INVITATION"] = "newSellerInvitation";
    ResendNotificationTemplates["NEW_ADMIN_INVITATION"] = "newAdminInvitation";
    ResendNotificationTemplates["SELLER_PRODUCT_APPROVED"] = "sellerProductApprovedEmailTemplate";
    ResendNotificationTemplates["SELLER_PRODUCT_REJECTED"] = "sellerProductRejectedEmailTemplate";
    ResendNotificationTemplates["ADMIN_REQUEST_CREATED"] = "adminRequestCreatedEmailTemplate";
    ResendNotificationTemplates["ADMIN_SELLER_REQUEST_CREATED"] = "adminSellerRequestCreatedEmailTemplate";
    ResendNotificationTemplates["SELLER_PAYOUT_SUMMARY"] = "sellerPayoutSummaryEmailTemplate";
})(ResendNotificationTemplates || (exports.ResendNotificationTemplates = ResendNotificationTemplates = {}));
