// @mercurjs/b2c-core/.medusa/server/src/workflows/seller/steps/create-seller-shipping-profile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createSellerShippingProfileStep = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const seller_1 = require("../../../modules/seller");
exports.createSellerShippingProfileStep = (0, workflows_sdk_1.createStep)("create-seller-shipping-profile", async ({ id: sellerId }, { container }) => {
    const link = container.resolve(utils_1.ContainerRegistrationKeys.LINK);
    const { result } = await core_flows_1.createShippingProfilesWorkflow.run({
        container,
        input: {
            data: [
                {
                    type: "default",
                    name: `${sellerId}:Default shipping profile`,
                },
            ],
        },
    });
    await link.create({
        [seller_1.SELLER_MODULE]: {
            seller_id: sellerId,
        },
        [utils_1.Modules.FULFILLMENT]: {
            shipping_profile_id: result[0].id,
        },
    });
});
