// @mercurjs/b2c-core/.medusa/server/src/workflows/seller/steps/send-invitation-email.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendSellerInvitationEmailStep = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const framework_1 = require("@mercurjs/framework");
const hosts_1 = require("../../../shared/infra/http/utils/hosts");
exports.sendSellerInvitationEmailStep = (0, workflows_sdk_1.createStep)("send-seller-invitation-email", async (input, { container }) => {
    const service = container.resolve(utils_1.Modules.NOTIFICATION);
    const logger = container.resolve(utils_1.ContainerRegistrationKeys.LOGGER);
    try {
        const storeData = await (0, framework_1.fetchStoreData)(container);
        const notification = await service.createNotifications({
            channel: "email",
            to: input.email,
            template: "newSellerInvitation",
            content: {
                subject: `You've been invited to join ${storeData.store_name}`,
            },
            data: {
                data: {
                    url: (0, hosts_1.buildHostAddress)(hosts_1.Hosts.VENDOR_PANEL, "/register"),
                    store_name: storeData.store_name,
                    storefront_url: storeData.storefront_url,
                },
            },
        });
        return new workflows_sdk_1.StepResponse(notification);
    }
    catch (e) {
        logger.error(e);
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.UNEXPECTED_STATE, "Notification provider failed!");
    }
});
