// @mercurjs/b2c-core/.medusa/server/src/api/vendor/products/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorProductsMiddlewares = void 0;
const multer_1 = __importDefault(require("multer"));
const framework_1 = require("@medusajs/framework");
const framework_2 = require("@mercurjs/framework");
const seller_product_1 = __importDefault(require("../../../links/seller-product"));
const middlewares_1 = require("../../../shared/infra/http/middlewares");
const middlewares_2 = require("../../../shared/infra/http/middlewares");
const query_config_1 = require("../attributes/query-config");
const validators_1 = require("../attributes/validators");
const query_config_2 = require("./query-config");
const validators_2 = require("./validators");
const canVendorCreateProduct = [
    (0, middlewares_2.checkConfigurationRule)(framework_2.ConfigurationRuleType.GLOBAL_PRODUCT_CATALOG, false),
    (0, middlewares_2.checkConfigurationRule)(framework_2.ConfigurationRuleType.PRODUCT_REQUEST_ENABLED, true),
];
const upload = (0, multer_1.default)({ storage: multer_1.default.memoryStorage() });
exports.vendorProductsMiddlewares = [
    {
        method: ["GET"],
        matcher: "/vendor/products",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.list),
            (0, middlewares_1.filterBySellerId)(),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products",
        middlewares: [
            ...canVendorCreateProduct,
            (0, framework_1.validateAndTransformBody)(validators_2.VendorCreateProduct),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/export",
        middlewares: [],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/import",
        middlewares: [
            (0, middlewares_2.checkConfigurationRule)(framework_2.ConfigurationRuleType.PRODUCT_IMPORT_ENABLED, true),
            upload.single("file"),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/products/:id",
        middlewares: [
            (0, framework_1.unlessPath)(/.*\/products\/(export|import)/, (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            })),
            (0, framework_1.unlessPath)(/.*\/products\/(export|import)/, (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve)),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id",
        middlewares: [
            (0, framework_1.unlessPath)(/.*\/products\/(export|import)/, (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            })),
            (0, framework_1.unlessPath)(/.*\/products\/(export|import)/, (0, framework_1.validateAndTransformBody)(validators_2.VendorUpdateProduct)),
            (0, framework_1.unlessPath)(/.*\/products\/(export|import)/, (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve)),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id/variants",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_2.CreateProductVariant),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id/variants/:variant_id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_2.UpdateProductVariant),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["DELETE"],
        matcher: "/vendor/products/:id/variants/:variant_id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id/options",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_2.CreateProductOption),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id/options/:option_id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_2.UpdateProductOption),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["DELETE"],
        matcher: "/vendor/products/:id/options/:option_id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
        ],
    },
    {
        method: ["DELETE"],
        matcher: "/vendor/products/:id",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/products/:id/status",
        middlewares: [
            (0, middlewares_1.checkResourceOwnershipByResourceId)({
                entryPoint: seller_product_1.default.entryPoint,
                filterField: "product_id",
            }),
            (0, framework_1.validateAndTransformBody)(validators_2.VendorUpdateProductStatus),
            (0, framework_1.validateAndTransformQuery)(validators_2.VendorGetProductParams, query_config_2.vendorProductQueryConfig.retrieve),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/products/:id/applicable-attributes",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetAttributesParams, query_config_1.retrieveAttributeQueryConfig),
        ],
    },
];
