// @mercurjs/b2c-core/.medusa/server/src/api/vendor/products/query-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorProductQueryConfig = exports.vendorProductFields = void 0;
exports.vendorProductFields = [
    "id",
    "title",
    "subtitle",
    "status",
    "external_id",
    "description",
    "handle",
    "is_giftcard",
    "discountable",
    "thumbnail",
    "collection_id",
    "type_id",
    "weight",
    "length",
    "height",
    "width",
    "hs_code",
    "origin_country",
    "mid_code",
    "material",
    "metadata",
    "*type",
    "*collection",
    "*options",
    "*options.values",
    "*tags",
    "*images",
    "*variants",
    "*variants.prices",
    "variants.prices.price_rules.value",
    "variants.prices.price_rules.attribute",
    "*variants.options",
    "*attribute_values",
    "*attribute_values.attribute",
    "*secondary_categories",
];
exports.vendorProductQueryConfig = {
    list: {
        defaults: exports.vendorProductFields,
        isList: true,
    },
    retrieve: {
        defaults: exports.vendorProductFields,
        isList: false,
    },
};
