// @mercurjs/b2c-core/.medusa/server/src/api/vendor/products/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = exports.GET = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const framework_1 = require("@mercurjs/framework");
const utils_2 = require("../../../shared/infra/http/utils");
const utils_3 = require("./utils");
/**
 * @oas [get] /vendor/products
 * operationId: "VendorListProducts"
 * summary: "List Products"
 * description: "Retrieves a list of products for the authenticated vendor."
 * x-authenticated: true
 * parameters:
 *   - name: offset
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to skip before starting to collect the result set.
 *   - name: limit
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to return.
 *   - name: fields
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: Comma-separated fields to include in the response.
 *   - name: order
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: The order of the returned items.
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             products:
 *               type: array
 *               items:
 *                 $ref: "#/components/schemas/VendorProduct"
 *             count:
 *               type: integer
 *               description: The total number of items available
 *             offset:
 *               type: integer
 *               description: The number of items skipped before these items
 *             limit:
 *               type: integer
 *               description: The number of items per page
 * tags:
 *   - Vendor Products
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const GET = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { productIds, count } = await (0, utils_3.filterProductsBySeller)(req.scope, req.filterableFields.seller_id, req.queryConfig.pagination?.skip || 0, req.queryConfig.pagination?.take || 10, req.filterableFields.sales_channel_id, req.queryConfig.pagination?.order);
    const { data: sellerProducts } = await query.graph({
        entity: "product",
        fields: req.queryConfig.fields,
        filters: {
            id: productIds,
        },
        pagination: {
            order: req.queryConfig.pagination?.order
        }
    });
    res.json({
        products: sellerProducts,
        count: count,
        offset: req.queryConfig.pagination?.skip || 0,
        limit: req.queryConfig.pagination?.take || 10,
        order: req.queryConfig.pagination?.order
    });
};
exports.GET = GET;
/**
 * @oas [post] /vendor/products
 * operationId: "VendorCreateProduct"
 * summary: "Create a Product"
 * description: "Creates a new product for the authenticated vendor."
 * x-authenticated: true
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateProduct"
 * responses:
 *   "201":
 *     description: Created
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             product:
 *               $ref: "#/components/schemas/VendorProduct"
 * tags:
 *   - Vendor Products
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const seller = await (0, utils_2.fetchSellerByAuthActorId)(req.auth_context?.actor_id, req.scope);
    const { additional_data, ...validatedBody } = req.validatedBody;
    const { result: [createdProduct], } = await core_flows_1.createProductsWorkflow.run({
        container: req.scope,
        input: {
            products: [
                {
                    ...validatedBody,
                    status: validatedBody.status === "draft" ? "draft" : "proposed",
                },
            ],
            additional_data: { ...additional_data, seller_id: seller.id },
        },
    });
    const eventBus = req.scope.resolve(utils_1.Modules.EVENT_BUS);
    await eventBus.emit({
        name: framework_1.ProductRequestUpdatedEvent.TO_CREATE,
        data: {
            seller_id: seller.id,
            data: {
                data: {
                    ...createdProduct,
                    product_id: createdProduct.id,
                },
                submitter_id: req.auth_context.actor_id,
                type: "product",
                status: createdProduct.status === "draft" ? "draft" : "pending",
            },
        },
    });
    const product_id = createdProduct.id;
    const { data: [product], } = await query.graph({
        entity: "product",
        fields: req.queryConfig.fields,
        filters: { id: product_id },
    }, { throwIfKeyNotFound: true });
    res.status(201).json({ product });
};
exports.POST = POST;
