// @mercurjs/requests/.medusa/server/src/subscribers/notification-admin-request-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = requestCreatedAdminNotifyHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
async function requestCreatedAdminNotifyHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const { data: { type }, } = event;
    await notificationService.createNotifications({
        to: "",
        channel: "feed",
        template: "admin-ui",
        content: {
            subject: `New Request Created`,
        },
        data: {
            title: `New ${framework_1.HumanizeTypes[type]}`,
            description: "You have received a new request from a seller 🔔",
            redirect: "/admin/requests",
        },
    });
}
exports.config = {
    event: framework_1.RequestUpdated.CREATED,
    context: {
        subscriberId: "request-created-admin-notify-handler",
    },
};
