// @mercurjs/requests/.medusa/server/src/subscribers/notification-seller-product-request-accepted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerProductRequestAcceptedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const requests_1 = require("../modules/requests");
async function sellerProductRequestAcceptedHandler({ event, container, }) {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    if (!(0, framework_1.checkConfigurationRule)(container, framework_1.ConfigurationRuleType.REQUIRE_PRODUCT_APPROVAL)) {
        return;
    }
    const { data: [productRequest], } = await query.graph({
        entity: "request",
        fields: ["*"],
        filters: {
            id: event.data.id,
        },
    });
    if (!productRequest || productRequest.type !== "product") {
        return;
    }
    const { data: [member], } = await query.graph({
        entity: "member",
        fields: ["*"],
        filters: {
            id: productRequest.submitter_id,
        },
    });
    if (!member || !member.email) {
        return;
    }
    await (0, requests_1.sendVendorUIRequestNotification)({
        container,
        requestId: event.data.id,
        requestType: "product",
        template: "seller_product_request_accepted_notification",
    });
}
exports.config = {
    event: framework_1.ProductRequestUpdatedEvent.ACCEPTED,
    context: {
        subscriberId: "seller-product-request-accepted-handler",
    },
};
