// @mercurjs/requests/.medusa/server/src/subscribers/product-update-request-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = productUpdateRequestCreatedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const framework_2 = require("@mercurjs/framework");
const requests_1 = require("../modules/requests");
const workflows_1 = require("../workflows/requests/workflows");
async function productUpdateRequestCreatedHandler({ event, container, }) {
    const { id } = event.data;
    const service = container.resolve(requests_1.REQUESTS_MODULE);
    const logger = container.resolve(utils_1.ContainerRegistrationKeys.LOGGER);
    const request = await service.retrieveRequest(id);
    if (request.status !== "draft" &&
        !(await (0, framework_1.checkConfigurationRule)(container, framework_1.ConfigurationRuleType.REQUIRE_PRODUCT_APPROVAL))) {
        logger.info(`${request.id}: Request automatically accepted`);
        await workflows_1.acceptProductRequestWorkflow.run({
            container,
            input: {
                data: request.data,
                id: request.id,
                reviewer_id: "system",
                reviewer_note: "auto accepted",
                status: "accepted",
            },
        });
    }
}
exports.config = {
    event: framework_2.ProductUpdateRequestUpdatedEvent.CREATED,
    context: {
        subscriberId: "product-update-request-created",
    },
};
