// @mercurjs/b2c-core/.medusa/server/src/workflows/fulfillment-set/create-location-fulfillment-set-and-associate-with-seller.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createLocationFulfillmentSetAndAssociateWithSellerWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const workflows_sdk_2 = require("@medusajs/workflows-sdk");
const seller_1 = require("../../modules/seller");
exports.createLocationFulfillmentSetAndAssociateWithSellerWorkflow = (0, workflows_sdk_2.createWorkflow)("create-fulfillment-set-and-associate-with-seller", function (input) {
    const fulfillmentSet = (0, core_flows_1.createFulfillmentSets)([
        {
            name: input.fulfillment_set_data.name,
            type: input.fulfillment_set_data.type,
        },
    ]);
    const fullfillmentSetId = (0, workflows_sdk_1.transform)(fulfillmentSet, (data) => data[0].id);
    (0, core_flows_1.createRemoteLinkStep)([
        {
            [utils_1.Modules.STOCK_LOCATION]: {
                stock_location_id: input.location_id,
            },
            [utils_1.Modules.FULFILLMENT]: {
                fulfillment_set_id: fullfillmentSetId,
            },
        },
        {
            [seller_1.SELLER_MODULE]: {
                seller_id: input.seller_id,
            },
            [utils_1.Modules.FULFILLMENT]: {
                fulfillment_set_id: fullfillmentSetId,
            },
        },
    ]);
});
