// @mercurjs/commission/.medusa/server/src/modules/commission/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("@medusajs/framework/utils");
const models_1 = require("./models");
const commission_line_1 = require("./models/commission_line");
class CommissionModuleService extends (0, utils_1.MedusaService)({
    CommissionRate: models_1.CommissionRate,
    CommissionRule: models_1.CommissionRule,
    CommissionLine: commission_line_1.CommissionLine,
}) {
    async selectCommissionRule(reference, reference_id) {
        const [rule] = await this.listCommissionRules({ reference, reference_id, is_active: true, deleted_at: null }, { relations: ["rate"] });
        return rule;
    }
    /**
     * Looks for first applicable CommissionRule for given context. The queries are executed in assumed priority order.
     * @param ctx Calculation context
     * @returns CommissionRule applicable for given context or null
     */
    async selectCommissionForProductLine(ctx) {
        const ruleQueries = [
            {
                reference: "seller+product_type",
                reference_id: `${ctx.seller_id}+${ctx.product_type_id}`,
            },
            {
                reference: "seller+product_category",
                reference_id: `${ctx.seller_id}+${ctx.product_category_id}`,
            },
            { reference: "seller", reference_id: ctx.seller_id },
            { reference: "product_type", reference_id: ctx.product_type_id },
            { reference: "product_category", reference_id: ctx.product_category_id },
            { reference: "site", reference_id: "" },
        ];
        for (const { reference, reference_id } of ruleQueries) {
            const rule = await this.selectCommissionRule(reference, reference_id);
            if (rule) {
                return rule;
            }
        }
        return null;
    }
}
exports.default = CommissionModuleService;
