// @mercurjs/b2c-core/.medusa/server/src/api/admin/attributes/query-config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAttributeValueQueryConfig = exports.retrieveAttributeValueQueryConfig = exports.defaultAdminAttributeValueFields = exports.listAttributeQueryConfig = exports.retrieveAttributeQueryConfig = exports.defaultAdminAttributeFields = void 0;
exports.defaultAdminAttributeFields = [
    'id',
    'name',
    'description',
    'handle',
    'is_filterable',
    'is_required',
    'ui_component',
    'metadata',
    '*possible_values',
    'product_categories.id',
    'product_categories.name'
];
exports.retrieveAttributeQueryConfig = {
    defaults: exports.defaultAdminAttributeFields,
    isList: false
};
exports.listAttributeQueryConfig = {
    ...exports.retrieveAttributeQueryConfig,
    defaultLimit: 50,
    isList: true
};
exports.defaultAdminAttributeValueFields = [
    'id',
    'value',
    'rank',
    'metadata'
];
exports.retrieveAttributeValueQueryConfig = {
    defaults: exports.defaultAdminAttributeValueFields,
    isList: false
};
exports.listAttributeValueQueryConfig = {
    ...exports.retrieveAttributeValueQueryConfig,
    isList: true,
    defaultLimit: 50
};
