// @mercurjs/b2c-core/.medusa/server/src/api/admin/attributes/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminCreateAttribute = exports.CreateAttribute = exports.AdminUpdateAttribute = exports.AdminUpdateAttributeValue = exports.AdminCreateAttributeValue = exports.AdminGetAttributesParams = exports.GetAttributesParams = exports.AdminGetAttributeParams = exports.AdminGetAttributeValuesParams = exports.AdminGetAttributeValueParams = void 0;
const zod_1 = require("zod");
const common_1 = require("@medusajs/medusa/api/utils/common-validators/common");
const validators_1 = require("@medusajs/medusa/api/utils/validators");
var AttributeUIComponent;
(function (AttributeUIComponent) {
    AttributeUIComponent["SELECT"] = "select";
    AttributeUIComponent["MULTIVALUE"] = "multivalue";
    AttributeUIComponent["UNIT"] = "unit";
    AttributeUIComponent["TOGGLE"] = "toggle";
    AttributeUIComponent["TEXTAREA"] = "text_area";
    AttributeUIComponent["COLOR_PICKER"] = "color_picker";
})(AttributeUIComponent || (AttributeUIComponent = {}));
exports.AdminGetAttributeValueParams = (0, validators_1.createSelectParams)();
exports.AdminGetAttributeValuesParams = (0, validators_1.createFindParams)();
exports.AdminGetAttributeParams = (0, validators_1.createSelectParams)();
exports.GetAttributesParams = zod_1.z.object({
    id: zod_1.z.string().optional(),
    name: zod_1.z.string().optional(),
    handle: zod_1.z.string().optional(),
    is_required: zod_1.z.boolean().optional(),
    is_filterable: zod_1.z.boolean().optional(),
    created_at: (0, validators_1.createOperatorMap)().optional(),
    updated_at: (0, validators_1.createOperatorMap)().optional(),
    deleted_at: (0, validators_1.createOperatorMap)().optional(),
    ui_component: zod_1.z.nativeEnum(AttributeUIComponent).optional(),
});
exports.AdminGetAttributesParams = (0, validators_1.createFindParams)({
    offset: 0,
    limit: 50,
})
    .merge((0, common_1.applyAndAndOrOperators)(exports.GetAttributesParams))
    .merge(exports.GetAttributesParams);
exports.AdminCreateAttributeValue = zod_1.z.object({
    value: zod_1.z.string().min(1),
    rank: zod_1.z.number(),
    metadata: zod_1.z.record(zod_1.z.unknown()).optional(),
});
exports.AdminUpdateAttributeValue = zod_1.z.object({
    id: zod_1.z.string().optional(),
    value: zod_1.z.string().optional(),
    rank: zod_1.z.number().optional(),
    metadata: zod_1.z.record(zod_1.z.unknown()).optional(),
});
exports.AdminUpdateAttribute = zod_1.z
    .object({
    name: zod_1.z.string().optional(),
    description: zod_1.z.string().optional(),
    handle: zod_1.z.string().optional(),
    is_filterable: zod_1.z.boolean().optional(),
    is_required: zod_1.z.boolean().optional(),
    metadata: zod_1.z.record(zod_1.z.unknown()).optional(),
    ui_component: zod_1.z.nativeEnum(AttributeUIComponent).optional(),
    product_category_ids: zod_1.z.array(zod_1.z.string()).optional(),
    possible_values: zod_1.z.array(exports.AdminUpdateAttributeValue).optional(),
})
    .strict();
exports.CreateAttribute = zod_1.z.object({
    name: zod_1.z.string().min(1),
    description: zod_1.z.string().optional(),
    is_filterable: zod_1.z.boolean().optional(),
    is_required: zod_1.z.boolean().optional(),
    ui_component: zod_1.z
        .nativeEnum(AttributeUIComponent)
        .default(AttributeUIComponent.SELECT),
    handle: zod_1.z.string().optional(),
    metadata: zod_1.z.record(zod_1.z.unknown()).optional(),
    possible_values: zod_1.z.array(exports.AdminCreateAttributeValue).optional(),
    product_category_ids: zod_1.z.array(zod_1.z.string()).optional(),
});
exports.AdminCreateAttribute = (0, validators_1.WithAdditionalData)(exports.CreateAttribute, (schema) => {
    return schema.refine((data) => data.ui_component !== AttributeUIComponent.SELECT ||
        (data.possible_values && data.possible_values.length > 0), {
        message: "Possible values are required when ui_component is SELECT",
        path: ["possible_values"],
    });
});
