// @mercurjs/b2c-core/.medusa/server/src/workflows/attribute/workflows/delete-attribute-value.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteAttributeValueWorkflow = exports.deleteAttributeValueWorkflowId = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const attribute_1 = require("../../../modules/attribute");
const product_attribute_value_1 = __importDefault(require("../../../links/product-attribute-value"));
const steps_1 = require("../steps");
exports.deleteAttributeValueWorkflowId = "delete-attribute-value";
exports.deleteAttributeValueWorkflow = (0, workflows_sdk_1.createWorkflow)(exports.deleteAttributeValueWorkflowId, (input) => {
    const normalizedInput = (0, workflows_sdk_1.transform)({ input }, ({ input }) => Array.isArray(input) ? input : [input]);
    const attributeValueProductQuery = (0, core_flows_1.useQueryGraphStep)({
        entity: product_attribute_value_1.default.entryPoint,
        fields: ["product_id", "attribute_value_id"],
        filters: {
            attribute_value_id: normalizedInput,
        },
    });
    const deleted = (0, steps_1.deleteAttributeValueStep)(normalizedInput);
    const links = (0, workflows_sdk_1.transform)({ attributeValueProductQuery }, ({ attributeValueProductQuery }) => {
        const { data } = attributeValueProductQuery;
        return data.map((element) => ({
            [utils_1.Modules.PRODUCT]: {
                product_id: element.product_id,
            },
            [attribute_1.ATTRIBUTE_MODULE]: {
                attribute_value_id: element.attribute_value_id,
            },
        }));
    });
    (0, core_flows_1.dismissRemoteLinkStep)(links);
    return new workflows_sdk_1.WorkflowResponse(deleted);
});
