// @mercurjs/b2c-core/.medusa/server/src/workflows/attribute/workflows/update-attributes.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateAttributesWorkflow = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const attribute_1 = require("../../../modules/attribute");
const category_attribute_1 = __importDefault(require("../../../links/category-attribute"));
const steps_1 = require("../steps");
const updateAttributesWorkflowId = "update-attributes";
exports.updateAttributesWorkflow = (0, workflows_sdk_1.createWorkflow)(updateAttributesWorkflowId, (input) => {
    const toUpdateInput = (0, workflows_sdk_1.transform)({ input }, ({ input: { attributes } }) => {
        return attributes.map((attribute) => ({
            ...attribute,
            product_category_ids: undefined,
        }));
    });
    const updatedAttributes = (0, steps_1.updateAttributesStep)(toUpdateInput);
    const attributesIdsWithCategories = (0, workflows_sdk_1.transform)({ input, updatedAttributes }, ({ input, updatedAttributes }) => {
        const updatedAttributeIds = updatedAttributes.map((attr) => attr.id);
        const attributeIdsWithoutCategories = input.attributes
            .filter((attr) => !attr.product_category_ids)
            .map((attr) => attr.id);
        return (0, utils_1.arrayDifference)(updatedAttributeIds, attributeIdsWithoutCategories);
    });
    const currentCategoriesLinksResult = (0, core_flows_1.useQueryGraphStep)({
        entity: category_attribute_1.default.entryPoint,
        fields: ["attribute_id", "product_category_id"],
        filters: {
            attribute_id: attributesIdsWithCategories,
        },
    });
    const currentCategoriesLinks = (0, workflows_sdk_1.transform)({ currentCategoriesLinksResult }, ({ currentCategoriesLinksResult }) => {
        return currentCategoriesLinksResult.data;
    });
    const toDeleteCategoriesLinks = (0, workflows_sdk_1.transform)({ currentCategoriesLinks }, ({ currentCategoriesLinks }) => {
        if (!currentCategoriesLinks.length) {
            return [];
        }
        return currentCategoriesLinks.map(({ attribute_id, product_category_id }) => ({
            [utils_1.Modules.PRODUCT]: {
                product_category_id,
            },
            [attribute_1.ATTRIBUTE_MODULE]: {
                attribute_id,
            },
        }));
    });
    (0, core_flows_1.dismissRemoteLinkStep)(toDeleteCategoriesLinks);
    const toCreateCategoryLinks = (0, workflows_sdk_1.transform)({ input }, ({ input: { attributes } }) => {
        return attributes
            .filter((attribute) => attribute.product_category_ids)
            .flatMap((attribute) => attribute.product_category_ids.map((attrCat) => ({
            [utils_1.Modules.PRODUCT]: {
                product_category_id: attrCat.id,
            },
            [attribute_1.ATTRIBUTE_MODULE]: {
                attribute_id: attribute.id,
            },
        })));
    });
    (0, core_flows_1.createRemoteLinkStep)(toCreateCategoryLinks);
    return new workflows_sdk_1.WorkflowResponse(updatedAttributes);
});
