// @mercurjs/b2c-core/.medusa/server/src/api/admin/notifications/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GET = void 0;
const http_1 = require("@medusajs/framework/http");
/**
 * @oas [get] /admin/notifications
 * operationId: "AdminListNotifications"
 * summary: "List Notifications"
 * description: "Retrieves a list of admin notifications from the feed channel."
 * x-authenticated: true
 * parameters:
 *   - name: offset
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to skip before starting to collect the result set.
 *   - name: limit
 *     in: query
 *     schema:
 *       type: number
 *     required: false
 *     description: The number of items to return.
 *   - name: fields
 *     in: query
 *     schema:
 *       type: string
 *     required: false
 *     description: Comma-separated fields to include in the response.
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             notifications:
 *               type: array
 *               items:
 *                 $ref: "#/components/schemas/AdminNotification"
 *             count:
 *               type: integer
 *               description: The total number of items available
 *             offset:
 *               type: integer
 *               description: The number of items skipped before these items
 *             limit:
 *               type: integer
 *               description: The number of items per page
 * tags:
 *   - Admin Notifications
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const GET = async (req, res) => {
    const { data: notifications, metadata } = await (0, http_1.refetchEntities)({
        entity: 'notification',
        idOrFilter: { ...req.filterableFields, channel: 'feed' },
        scope: req.scope,
        fields: req.queryConfig.fields,
        pagination: req.queryConfig.pagination
    });
    res.json({
        notifications,
        count: metadata.count,
        offset: metadata.skip,
        limit: metadata.take
    });
};
exports.GET = GET;
