// @mercurjs/b2c-core/.medusa/server/src/shared/infra/http/utils/products.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fetchProductDetails = void 0;
exports.getApplicableAttributes = getApplicableAttributes;
const utils_1 = require("@medusajs/framework/utils");
const category_attribute_1 = __importDefault(require("../../../../links/category-attribute"));
const fetchProductDetails = async (product_id, scope) => {
    const service = scope.resolve(utils_1.Modules.PRODUCT);
    const product = await service.retrieveProduct(product_id);
    return product;
};
exports.fetchProductDetails = fetchProductDetails;
async function getApplicableAttributes(container, product_id, fields) {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [product] } = await query.graph({
        entity: 'product',
        fields: ['categories.id'],
        filters: {
            id: product_id
        }
    });
    const categoryIds = product.categories.map((category) => category.id);
    const { data: attributes } = await query.graph({
        entity: category_attribute_1.default.entryPoint,
        fields: ['attribute_id']
    });
    const attributeIds = attributes.map((attribute) => attribute.attribute_id);
    const { data: globalAttributes } = await query.graph({
        entity: 'attribute',
        fields: fields,
        filters: {
            id: {
                $nin: attributeIds
            }
        }
    });
    const { data: categoryAttributes } = await query.graph({
        entity: category_attribute_1.default.entryPoint,
        fields: fields.map((field) => `attribute.${field}`),
        filters: {
            product_category_id: categoryIds
        }
    });
    return [
        ...globalAttributes,
        ...categoryAttributes.map((rel) => rel.attribute)
    ];
}
