// @mercurjs/b2c-core/.medusa/server/src/workflows/attribute/steps/validate-attribute-value.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateAttributeValueStep = exports.validateAttributeValueStepId = void 0;
const utils_1 = require("@medusajs/framework/utils");
const workflows_sdk_1 = require("@medusajs/framework/workflows-sdk");
const product_attribute_value_1 = __importDefault(require("../../../links/product-attribute-value"));
exports.validateAttributeValueStepId = "validate-attribute-value";
exports.validateAttributeValueStep = (0, workflows_sdk_1.createStep)(exports.validateAttributeValueStepId, async (input, { container }) => {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [attribute], } = await query.graph({
        entity: "attribute",
        fields: ["product_categories.id", "possible_values.value"],
        filters: {
            id: input.attribute_id,
        },
    });
    const allowedValues = attribute.possible_values?.map((posVal) => posVal.value);
    if (allowedValues?.length && !allowedValues.includes(input.value)) {
        throw new utils_1.MedusaError(utils_1.MedusaErrorTypes.INVALID_DATA, `Attribute ${input.attribute_id} doesn't define ${input.value} as a possible_value`);
    }
    const attributeCategoryIds = attribute.product_categories.map((cat) => cat.id);
    // If all attributes are global, we don't enforce for product.categories to include the attribute.product_categories, since there are none
    if (attributeCategoryIds.length) {
        const { data: [product], } = await query.graph({
            entity: "product",
            fields: ["categories.id"],
            filters: {
                id: input.product_id,
            },
        });
        const productCategoryIds = product.categories?.map((cat) => cat.id);
        if (!productCategoryIds?.some((prodCatId) => attributeCategoryIds.includes(prodCatId))) {
            throw new utils_1.MedusaError(utils_1.MedusaErrorTypes.INVALID_DATA, `Product ${input.product_id} isn't linked to any category from the requested attributes.`);
        }
    }
    const { data: attributeValuesProduct } = await query.graph({
        entity: product_attribute_value_1.default.entryPoint,
        fields: ["attribute_value.value", "attribute_value.attribute_id"],
        filters: {
            product_id: input.product_id,
        },
    });
    const attributeValues = attributeValuesProduct.map((element) => element.attribute_value);
    if (attributeValues.some((value) => value.attribute_id === input.attribute_id &&
        value.value === input.value)) {
        throw new utils_1.MedusaError(utils_1.MedusaErrorTypes.DUPLICATE_ERROR, `Attribute value ${input.value} for attribute ${input.attribute_id} already exists for product ${input.product_id}`);
    }
    return new workflows_sdk_1.StepResponse();
});
