// @mercurjs/commission/.medusa/server/src/subscribers/commission-order-set-placed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = commissionOrderSetPlacedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const workflows_1 = require("../workflows/commission/workflows");
async function commissionOrderSetPlacedHandler({ event, container, }) {
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [set], } = await query.graph({
        entity: "order_set",
        fields: ["orders.id"],
        filters: {
            id: event.data.id,
        },
    });
    const ordersCreated = set.orders.map((o) => o.id);
    for (const order_id of ordersCreated) {
        const { data: [seller], } = await query.graph({
            entity: framework_1.SELLER_ORDER_LINK,
            fields: ["seller_id"],
            filters: {
                order_id: order_id,
            },
        });
        if (!seller) {
            return;
        }
        await workflows_1.calculateCommissionWorkflow.run({
            input: {
                order_id: order_id,
                seller_id: seller.seller_id,
            },
            container,
        });
    }
}
exports.config = {
    event: framework_1.OrderSetWorkflowEvents.PLACED,
    context: {
        subscriberId: "commission-order-set-placed-handler",
    },
};
