// @mercurjs/b2c-core/.medusa/server/src/api/store/carts/[id]/promotions/route.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const framework_1 = require("@mercurjs/framework");
const seller_1 = require("../../../../../modules/seller");
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const seller = await (0, framework_1.fetchSellerByAuthActorId)(req.auth_context?.actor_id, req.scope);
    const validatePromotions = await (0, seller_1.validateSellerPromotions)(req.validatedBody.promo_codes, req.scope, seller.id);
    if (!validatePromotions.valid) {
        throw new utils_1.MedusaError(utils_1.MedusaError.Types.INVALID_DATA, 'Some of the promotion codes are invalid');
    }
    await core_flows_1.updateCartPromotionsWorkflow.run({
        input: {
            cart_id: req.params.id,
            promo_codes: req.validatedBody.promo_codes,
            action: req.validatedBody.promo_codes.length > 0
                ? utils_1.PromotionActions.ADD
                : utils_1.PromotionActions.REPLACE,
        },
    });
    const { data: [cart] } = await query.graph({
        entity: 'cart',
        filters: {
            id: req.params.id
        },
        fields: req.queryConfig.fields
    });
    res.json({ cart });
};
exports.POST = POST;
