// @mercurjs/b2c-core/.medusa/server/src/api/vendor/shipping-profiles/route.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GET = exports.POST = void 0;
const utils_1 = require("@medusajs/framework/utils");
const core_flows_1 = require("@medusajs/medusa/core-flows");
const seller_1 = require("../../../modules/seller");
const seller_shipping_profile_1 = __importDefault(require("../../../links/seller-shipping-profile"));
const utils_2 = require("../../../shared/infra/http/utils");
/**
 * @oas [post] /vendor/shipping-profiles
 * operationId: "VendorCreateShippingProfile"
 * summary: "Create a Shipping profile"
 * description: "Creates a Shipping profile."
 * x-authenticated: true
 * parameters:
 *   - in: query
 *     name: fields
 *     description: The comma-separated fields to include in the response
 *     schema:
 *       type: string
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/VendorCreateShippingProfile"
 * responses:
 *   "201":
 *     description: Created
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             shipping_profile:
 *               $ref: "#/components/schemas/VendorShippingProfile"
 * tags:
 *   - Vendor Shipping Profiles
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const POST = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const link = req.scope.resolve(utils_1.ContainerRegistrationKeys.LINK);
    const seller = await (0, utils_2.fetchSellerByAuthActorId)(req.auth_context.actor_id, req.scope);
    const { result } = await core_flows_1.createShippingProfilesWorkflow.run({
        container: req.scope,
        input: {
            data: [
                {
                    type: req.validatedBody.type,
                    name: `${seller.id}:${req.validatedBody.name}`,
                },
            ],
        },
    });
    const { data: [shipping_profile], } = await query.graph({
        entity: "shipping_profile",
        fields: req.queryConfig.fields,
        filters: {
            id: result[0].id,
        },
    });
    await link.create({
        [seller_1.SELLER_MODULE]: {
            seller_id: seller.id,
        },
        [utils_1.Modules.FULFILLMENT]: {
            shipping_profile_id: shipping_profile.id,
        },
    });
    res.status(201).json({ shipping_profile });
};
exports.POST = POST;
/**
 * @oas [get] /vendor/shipping-profiles
 * operationId: "VendorListShippingProfiles"
 * summary: "List shipping profiles"
 * description: "Retrieves a list of shipping profiles."
 * x-authenticated: true
 * parameters:
 *   - in: query
 *     name: fields
 *     description: The comma-separated fields to include in the response
 *     schema:
 *       type: string
 * responses:
 *   "200":
 *     description: OK
 *     content:
 *       application/json:
 *         schema:
 *           type: object
 *           properties:
 *             shipping_profiles:
 *               type: array
 *               items:
 *                 $ref: "#/components/schemas/VendorShippingProfile"
 * tags:
 *   - Vendor Shipping Profiles
 * security:
 *   - api_token: []
 *   - cookie_auth: []
 */
const GET = async (req, res) => {
    const query = req.scope.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: shipping_profiles, metadata } = await query.graph({
        entity: seller_shipping_profile_1.default.entryPoint,
        fields: req.queryConfig.fields.map((field) => `shipping_profile.${field}`),
        filters: {
            ...req.filterableFields,
            deleted_at: {
                $eq: null,
            },
        },
        pagination: req.queryConfig.pagination,
    });
    res.json({
        shipping_profiles,
        count: metadata?.count,
        offset: metadata?.skip,
        limit: metadata?.take,
    });
};
exports.GET = GET;
