// @mercurjs/requests/.medusa/server/src/api/vendor/return-request/middlewares.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.vendorReturnRequestsMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const seller_return_request_1 = __importDefault(require("../../../links/seller-return-request"));
const query_config_1 = require("./query-config");
const validators_1 = require("./validators");
const framework_2 = require("@mercurjs/framework");
exports.vendorReturnRequestsMiddlewares = [
    {
        method: ["GET"],
        matcher: "/vendor/return-request",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetOrderReturnRequestParams, query_config_1.vendorReturnOrderRequestQueryConfig.list),
            (0, framework_2.filterBySellerId)(),
        ],
    },
    {
        method: ["GET"],
        matcher: "/vendor/return-request/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetOrderReturnRequestParams, query_config_1.vendorReturnOrderRequestQueryConfig.retrieve),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_return_request_1.default.entryPoint,
                filterField: "order_return_request_id",
            }),
        ],
    },
    {
        method: ["POST"],
        matcher: "/vendor/return-request/:id",
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.VendorGetOrderReturnRequestParams, query_config_1.vendorReturnOrderRequestQueryConfig.retrieve),
            (0, framework_1.validateAndTransformBody)(validators_1.VendorUpdateOrderReturnRequest),
            (0, framework_2.checkResourceOwnershipByResourceId)({
                entryPoint: seller_return_request_1.default.entryPoint,
                filterField: "order_return_request_id",
            }),
        ],
    },
];
