// @mercurjs/b2c-core/.medusa/server/src/subscribers/seller-creation-request-accepted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerCreationRequestAcceptedHandler;
const framework_1 = require("@mercurjs/framework");
const workflows_1 = require("../workflows");
const utils_1 = require("@medusajs/framework/utils");
async function sellerCreationRequestAcceptedHandler({ event, container, }) {
    const logger = container.resolve(utils_1.ContainerRegistrationKeys.LOGGER);
    const request = event.data;
    const { result: seller } = await workflows_1.createSellerWorkflow.run({
        container,
        input: {
            member: request.data.member,
            seller: request.data.seller,
            auth_identity_id: request.data.auth_identity_id,
        },
    });
    logger.info(`Seller creation request accepted: ${request.id}, seller: ${seller.id}`);
}
exports.config = {
    event: framework_1.SellerAccountRequestUpdatedEvent.ACCEPTED,
    context: {
        subscriberId: "seller-creation-request-accepted-handler",
    },
};
