// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-cancel-order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerCancelOrderHandler;
const utils_1 = require("@medusajs/framework/utils");
const resend_1 = require("../providers/resend");
const framework_1 = require("@mercurjs/framework");
async function sellerCancelOrderHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [order], } = await query.graph({
        entity: "order",
        fields: [
            "id",
            "email",
            "display_id",
            "items.*",
            "customer.first_name",
            "customer.last_name",
            "seller.*",
        ],
        filters: {
            id: event.data.id,
        },
    });
    if (!order) {
        console.error("Order not found:", event.data.id);
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: order.seller.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_CANCELED_ORDER,
        content: {
            subject: `Your order #${order.display_id} has been canceled`,
        },
        data: {
            data: {
                order: {
                    id: order.id,
                    display_id: order.display_id,
                    item: order.items,
                },
                order_address: (0, framework_1.buildHostAddress)(framework_1.Hosts.VENDOR_PANEL, `/orders/${order.id}`).toString(),
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: "order.canceled",
    context: {
        subscriberId: "notification-seller-cancel-order-resend",
    },
};
