// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-account-request-accepted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerRequestAcceptedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function sellerRequestAcceptedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const requestData = event.data.data;
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: requestData.provider_identity_id,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_ACCOUNT_UPDATES_APPROVAL,
        content: {
            subject: `${storeData.store_name} - Seller account approved!`,
        },
        data: { data: { user_name: requestData.member.name, store_name: storeData.store_name, storefront_url: storeData.storefront_url } },
    });
}
exports.config = {
    event: framework_1.SellerAccountRequestUpdatedEvent.ACCEPTED,
    context: {
        subscriberId: "seller-account-request-accepted-handler-resend",
    },
};
