// @mercurjs/resend/.medusa/server/src/subscribers/notification-admin-request-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = requestCreatedAdminNotifyHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function requestCreatedAdminNotifyHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const { data: { type, data }, } = event;
    if (type === "seller") {
        const admins = await (0, framework_1.fetchAdminEmails)(container);
        const storeData = await (0, framework_1.fetchStoreData)(container);
        const notifications = admins.map((email) => ({
            to: email,
            channel: "email",
            template: resend_1.ResendNotificationTemplates.ADMIN_SELLER_REQUEST_CREATED,
            content: {
                subject: "Seller requested to join the platform",
            },
            data: {
                data: {
                    seller_name: data.seller.name || "",
                    request_address: (0, framework_1.buildHostAddress)(framework_1.Hosts.BACKEND, `/admin/requests/seller`).toString(),
                    store_name: storeData.store_name,
                    storefront_url: storeData.storefront_url,
                },
            },
        }));
        await notificationService.createNotifications(notifications);
    }
}
exports.config = {
    event: framework_1.RequestUpdated.CREATED,
    context: {
        subscriberId: "request-created-admin-notify-handler-resend",
    },
};
