// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-account-request-rejected.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerRequestRejectedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function sellerRequestRejectedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const requestData = event.data.data;
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: requestData.provider_identity_id,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_ACCOUNT_UPDATES_REJECTION,
        content: {
            subject: `${storeData.store_name} - Seller account rejected!`,
        },
        data: {
            data: {
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: framework_1.SellerAccountRequestUpdatedEvent.REJECTED,
    context: {
        subscriberId: "seller-account-request-rejected-handler-resend",
    },
};
