// @mercurjs/resend/.medusa/server/src/subscribers/notification-admin-user-invite-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = adminUserInviteCreatedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function adminUserInviteCreatedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const userService = container.resolve(utils_1.Modules.USER);
    const invites = await userService.listInvites({
        id: event.data.id,
    });
    const storeData = await (0, framework_1.fetchStoreData)(container);
    const notifications = invites.map((invite) => ({
        to: invite.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.NEW_ADMIN_INVITATION,
        content: {
            subject: "Admin requested to join the platform",
        },
        data: {
            data: {
                url: (0, framework_1.buildHostAddress)(framework_1.Hosts.BACKEND, `/app/invite?token=${invite.token}`).toString(),
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    }));
    await notificationService.createNotifications(notifications);
}
exports.config = {
    event: utils_1.InviteWorkflowEvents.CREATED,
    context: {
        subscriberId: "admin-user-invite-created",
    },
};
