// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-payout-summary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = notificationSellerPayoutSummary;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function notificationSellerPayoutSummary({ event, container, }) {
    const { seller, payouts } = event.data;
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications([
        {
            to: seller.email,
            channel: "email",
            template: resend_1.ResendNotificationTemplates.SELLER_PAYOUT_SUMMARY,
            content: {
                subject: `Payout summary for ${seller.name}`,
            },
            data: {
                data: {
                    seller,
                    payouts,
                    store_name: storeData.store_name,
                    storefront_url: storeData.storefront_url,
                },
            },
        },
    ]);
}
exports.config = {
    event: framework_1.PayoutSummaryEvents.NOTIFICATION_SENT,
    context: {
        subscriberId: "notification-seller-payout-summary-resend",
    },
};
