// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-new-order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerNewOrderHandler;
const utils_1 = require("@medusajs/framework/utils");
const resend_1 = require("../providers/resend");
const framework_1 = require("@mercurjs/framework");
async function sellerNewOrderHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [order], } = await query.graph({
        entity: "order",
        fields: [
            "id",
            "display_id",
            "items.*",
            "seller.email",
            "seller.name",
            "seller.id",
            "customer.first_name",
            "customer.last_name",
        ],
        filters: {
            id: event.data.id,
        },
    });
    if (!order) {
        console.error("Order not found:", event.data.id);
        return;
    }
    const sellerEmail = order.seller?.email;
    if (!sellerEmail) {
        console.error("Seller email not found for order:", order.id);
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    const customer_name = `${order.customer?.first_name || ""} ${order.customer?.last_name || ""}`;
    await notificationService.createNotifications([
        {
            to: order.seller?.id,
            channel: "seller_feed",
            template: "seller_new_order_notification",
            data: {
                order_id: order.id,
                customer_name,
            },
        },
        {
            to: sellerEmail,
            channel: "email",
            template: resend_1.ResendNotificationTemplates.SELLER_NEW_ORDER,
            content: {
                subject: `New order #${order.display_id} received`,
            },
            data: {
                data: {
                    order_id: order.id,
                    order,
                    customer_name,
                    seller_name: order.seller?.name || "",
                    store_name: storeData.store_name,
                    storefront_url: storeData.storefront_url,
                },
            },
        },
    ]);
}
exports.config = {
    event: utils_1.OrderWorkflowEvents.PLACED,
    context: {
        subscriberId: "seller-new-order-handler-resend",
    },
};
