// @mercurjs/resend/.medusa/server/src/subscribers/notification-buyer-account-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = buyerAccountCreatedHandler;
const utils_1 = require("@medusajs/framework/utils");
const resend_1 = require("../providers/resend");
const framework_1 = require("@mercurjs/framework");
async function buyerAccountCreatedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [customer], } = await query.graph({
        entity: "customer",
        fields: ["id", "email", "first_name", "last_name"],
        filters: {
            id: event.data.id,
        },
    });
    if (!customer) {
        console.error("Customer not found:", event.data.id);
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: customer.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.BUYER_ACCOUNT_CREATED,
        content: {
            subject: `Welcome to ${storeData.store_name}, ${customer.first_name || ""}!`,
        },
        data: {
            data: {
                user_name: customer.first_name || "Customer",
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: "customer.created",
    context: {
        subscriberId: "buyer-account-created-handler-resend",
    },
};
