// @mercurjs/resend/.medusa/server/src/subscribers/notification-all-password-reset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = passwordResetHandler;
const utils_1 = require("@medusajs/framework/utils");
const resend_1 = require("../providers/resend");
const framework_1 = require("@mercurjs/framework");
async function passwordResetHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const hostType = framework_1.actorTypeToHost[event.data.actor_type];
    if (!hostType) {
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: event.data.entity_id,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.FORGOT_PASSWORD,
        content: {
            subject: `${storeData.store_name} - Reset password request`,
        },
        data: {
            data: {
                url: (0, framework_1.buildResetPasswordUrl)(hostType, event.data.token).toString(),
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: utils_1.AuthWorkflowEvents.PASSWORD_RESET,
    context: {
        subscriberId: "password-reset-handler-resend",
    },
};
