// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-seller-request-created.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = requestCreatedSellerAccountUpdatesNotifyHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function requestCreatedSellerAccountUpdatesNotifyHandler({ event, container, }) {
    const requestData = event.data.data;
    if (event.data.type !== "seller") {
        return;
    }
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: requestData.provider_identity_id,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_ACCOUNT_UPDATES_SUBMISSION,
        content: {
            subject: "Your submission has been received",
        },
        data: {
            data: {
                user_name: requestData.member.name,
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: framework_1.SellerRequest.CREATED,
    context: {
        subscriberId: "request-created-seller-account-updates-notify-handler-resend",
    },
};
