// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-product-request-accepted.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerProductRequestAcceptedHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
async function sellerProductRequestAcceptedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [productRequest], } = await query.graph({
        entity: "request",
        fields: ["*"],
        filters: {
            id: event.data.id,
        },
    });
    if (!productRequest || productRequest.type !== "product") {
        return;
    }
    if (productRequest.reviewer_id === "system") {
        return;
    }
    const { data: [member], } = await query.graph({
        entity: "member",
        fields: ["*"],
        filters: {
            id: productRequest.submitter_id,
        },
    });
    if (!member || !member.email) {
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: member.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_PRODUCT_APPROVED,
        content: {
            subject: `${storeData.store_name} - Product approved!`,
        },
        data: { data: { product_title: productRequest.data.title, store_name: storeData.store_name, storefront_url: storeData.storefront_url } },
    });
}
exports.config = {
    event: framework_1.ProductRequestUpdatedEvent.ACCEPTED,
    context: {
        subscriberId: "seller-product-request-accepted-handler-resend",
    },
};
