// @mercurjs/resend/.medusa/server/src/subscribers/notification-buyer-new-order.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = orderCreatedHandler;
const utils_1 = require("@medusajs/framework/utils");
const resend_1 = require("../providers/resend");
const framework_1 = require("@mercurjs/framework");
async function orderCreatedHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const query = container.resolve(utils_1.ContainerRegistrationKeys.QUERY);
    const { data: [order], } = await query.graph({
        entity: "order",
        fields: [
            "*",
            "customer.*",
            "items.*",
            "shipping_address.*",
            "shipping_methods.*",
            "summary.*",
            "order_set.*"
        ],
        filters: {
            id: event.data.id,
        },
    });
    if (!order) {
        return;
    }
    const storeData = await (0, framework_1.fetchStoreData)(container);
    const orderUrl = (0, framework_1.buildHostAddress)(framework_1.Hosts.STOREFRONT, `/user/orders/${order.order_set.id ?? order.id}`).toString();
    await notificationService.createNotifications({
        to: order.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.BUYER_NEW_ORDER,
        content: {
            subject: `Order Confirmation - #${order.display_id}`,
        },
        data: {
            data: {
                user_name: order.customer?.first_name || "Customer",
                order_id: order.id,
                order_address: orderUrl,
                order: {
                    ...order,
                    display_id: order.display_id,
                    total: order.summary?.current_order_total || 0,
                },
                store_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: utils_1.OrderWorkflowEvents.PLACED,
    context: {
        subscriberId: "order-created-handler-resend",
    },
};
