// @mercurjs/resend/.medusa/server/src/subscribers/notification-seller-team-invite.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.config = void 0;
exports.default = sellerTeamInviteHandler;
const utils_1 = require("@medusajs/framework/utils");
const framework_1 = require("@mercurjs/framework");
const resend_1 = require("../providers/resend");
const framework_2 = require("@mercurjs/framework");
async function sellerTeamInviteHandler({ event, container, }) {
    const notificationService = container.resolve(utils_1.Modules.NOTIFICATION);
    const invite = event.data;
    const storeData = await (0, framework_1.fetchStoreData)(container);
    await notificationService.createNotifications({
        to: invite.email,
        channel: "email",
        template: resend_1.ResendNotificationTemplates.SELLER_TEAM_MEMBER_INVITATION,
        content: {
            subject: `You've been invited to join a team on ${storeData.store_name}`,
        },
        data: {
            data: {
                user_name: invite.user_name,
                store_name: invite.store_name,
                host: (0, framework_2.buildInviteUrl)(invite.token).toString(),
                id: invite.id,
                email: invite.email,
                marketplace_name: storeData.store_name,
                storefront_url: storeData.storefront_url,
            },
        },
    });
}
exports.config = {
    event: framework_1.SellerTeamInviteEvent.CREATED,
    context: {
        subscriberId: "seller-team-invite-handler-resend",
    },
};
