// @mercurjs/requests/.medusa/server/src/api/store/return-request/middlewares.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.storeOrderReturnRequestsMiddlewares = void 0;
const framework_1 = require("@medusajs/framework");
const framework_2 = require("@mercurjs/framework");
const query_config_1 = require("./query-config");
const validators_1 = require("./validators");
exports.storeOrderReturnRequestsMiddlewares = [
    {
        method: ['GET'],
        matcher: '/store/return-request',
        middlewares: [
            (0, framework_1.validateAndTransformQuery)(validators_1.StoreGetOrderReturnRequestParams, query_config_1.storeReturnOrderRequestQueryConfig.list)
        ]
    },
    {
        method: ['POST'],
        matcher: '/store/return-request',
        middlewares: [
            (0, framework_1.validateAndTransformBody)(validators_1.StoreCreateReturnRequest),
            (0, framework_2.checkCustomerResourceOwnershipByResourceId)({
                entryPoint: 'order',
                resourceId: (req) => {
                    return req.validatedBody.order_id;
                }
            })
        ]
    },
    {
        method: ['GET'],
        matcher: '/store/return-request/:id',
        middlewares: [
            (0, framework_2.checkCustomerResourceOwnershipByResourceId)({
                entryPoint: 'order_return_request'
            }),
            (0, framework_1.validateAndTransformQuery)(validators_1.StoreGetOrderReturnRequestParams, query_config_1.storeReturnOrderRequestQueryConfig.retrieve)
        ]
    }
];
